<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>R6 class in which data sets will be stored for visualization in Cerebro. — Cerebro_v1.3 • cerebroApp</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="R6 class in which data sets will be stored for visualization in Cerebro. — Cerebro_v1.3" />
<meta property="og:description" content="A Cerebro_v1.3 object is an R6 class that contains several types of
data that can be visualized in Cerebro." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cerebroApp</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/cerebroApp_workflow_Seurat.html">The cerebroApp workflow (Seurat)</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">The Cerebro interface</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/Cerebro_interface_v1.2.html">Cerebro v1.2 and older</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="../articles/export_a_data_set_in_SCE_format.html">Export an SCE/SingleCellExperiment object</a>
    </li>
    <li>
      <a href="../articles/export_and_visualize_custom_tables_and_plots.html">Export and visualize custom tables and plots</a>
    </li>
    <li>
      <a href="../articles/launch_cerebro_with_pre-loaded_data_set.html">Launch Cerebro with pre-loaded data set</a>
    </li>
    <li>
      <a href="../articles/host_cerebro_on_shinyapps.html">Host Cerebro on shinyapps.io</a>
    </li>
    <li>
      <a href="../articles/control_access_to_cerebro_with_a_login_page.html">Control access to Cerebro with a login page</a>
    </li>
    <li>
      <a href="../articles/overview_of_cerebro_v1.3_class.html">Overview of the Cerebro v1.3 class</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Release notes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/release_notes_v1.3.html">v1.3</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/romanhaa/cerebroApp/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/fakechek1">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>R6 class in which data sets will be stored for visualization in Cerebro.</h1>
    <small class="dont-index">Source: <a href='https://github.com/romanhaa/cerebroApp/blob/master/R/class-Cerebro_v1.3.R'><code>R/class-Cerebro_v1.3.R</code></a></small>
    <div class="hidden name"><code>Cerebro_v1.3.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A <code>Cerebro_v1.3</code> object is an R6 class that contains several types of
data that can be visualized in Cerebro.</p>
    </div>



    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A new <code>Cerebro_v1.3</code> object.</p>
    <h2 class="hasAnchor" id="public-fields"><a class="anchor" href="#public-fields"></a>Public fields</h2>

    <p><div class="r6-fields"></p><dl>
<dt><code>version</code></dt><dd><p>cerebroApp version that was used to create the object.</p></dd>

<dt><code>experiment</code></dt><dd><p><code>list</code> that contains meta data about the data set,
including experiment name, species, date of export.</p></dd>

<dt><code>technical_info</code></dt><dd><p><code>list</code> that contains technical information
about the analysis, including the R session info.</p></dd>

<dt><code>parameters</code></dt><dd><p><code>list</code> that contains important parameters that
were used during the analysis, e.g. cut-off values for cell filtering.</p></dd>

<dt><code>groups</code></dt><dd><p><code>list</code> that contains specified grouping variables and
and the group levels (subgroups) that belong to each of them. For each
grouping variable, a corresponding column with the same name must exist
in the meta data.</p></dd>

<dt><code>cell_cycle</code></dt><dd><p><code>vector</code> that contains the name of columns in the
meta data that contain cell cycle assignments.</p></dd>

<dt><code>gene_lists</code></dt><dd><p><code>list</code> that contains gene lists, e.g.
mitochondrial and/or ribosomal genes.</p></dd>

<dt><code>expression</code></dt><dd><p><code>matrix</code>-like object that holds transcript counts.</p></dd>

<dt><code>meta_data</code></dt><dd><p><code>data.frame</code> that contains cell meta data.</p></dd>

<dt><code>projections</code></dt><dd><p><code>list</code> that contains projections/dimensional
reductions.</p></dd>

<dt><code>most_expressed_genes</code></dt><dd><p><code>list</code> that contains a <code>data.frame</code>
holding the most expressed genes for each grouping variable that was
specified during the call to <code><a href='getMostExpressedGenes.html'>getMostExpressedGenes</a></code>.</p></dd>

<dt><code>marker_genes</code></dt><dd><p><code>list</code> that contains a <code>list</code> for every
method that was used to calculate marker genes, and a <code>data.frame</code>
for each grouping variable, e.g. those that were specified during the
call to <code><a href='getMarkerGenes.html'>getMarkerGenes</a></code>.</p></dd>

<dt><code>enriched_pathways</code></dt><dd><p><code>list</code> that contains a <code>list</code> for
every method that was used to calculate marker genes, and a
<code>data.frame</code> for each grouping variable, e.g. those that were
specified during the call to <code><a href='getEnrichedPathways.html'>getEnrichedPathways</a></code> or
<code><a href='performGeneSetEnrichmentAnalysis.html'>performGeneSetEnrichmentAnalysis</a></code>.</p></dd>

<dt><code>trees</code></dt><dd><p><code>list</code> that contains a phylogenetic tree (class
<code>phylo</code>) for grouping variables.</p></dd>

<dt><code>trajectories</code></dt><dd><p><code>list</code> that contains a <code>list</code> for every
method that was used to calculate trajectories, and, depending on the
method, a <code>data.frame</code> or <code>list</code> for each specific trajectory,
e.g. those extracted with <code><a href='extractMonocleTrajectory.html'>extractMonocleTrajectory</a></code>.</p></dd>

<dt><code>extra_material</code></dt><dd><p><code>list</code> that can contain additional material
related to the data set; tables should be stored in <code>data.frame</code>
format in a named <code>list</code> called `tables`</p></dd>

</dl><p></div></p>
    <h2 class="hasAnchor" id="methods"><a class="anchor" href="#methods"></a>Methods</h2>

    
<h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Public methods</h3>

<ul>
<li><p><a href='#method-new'><code>Cerebro_v1.3$new()</code></a></p></li>
<li><p><a href='#method-setVersion'><code>Cerebro_v1.3$setVersion()</code></a></p></li>
<li><p><a href='#method-getVersion'><code>Cerebro_v1.3$getVersion()</code></a></p></li>
<li><p><a href='#method-checkIfGroupExists'><code>Cerebro_v1.3$checkIfGroupExists()</code></a></p></li>
<li><p><a href='#method-checkIfColumnExistsInMetadata'><code>Cerebro_v1.3$checkIfColumnExistsInMetadata()</code></a></p></li>
<li><p><a href='#method-addExperiment'><code>Cerebro_v1.3$addExperiment()</code></a></p></li>
<li><p><a href='#method-getExperiment'><code>Cerebro_v1.3$getExperiment()</code></a></p></li>
<li><p><a href='#method-addParameters'><code>Cerebro_v1.3$addParameters()</code></a></p></li>
<li><p><a href='#method-getParameters'><code>Cerebro_v1.3$getParameters()</code></a></p></li>
<li><p><a href='#method-addTechnicalInfo'><code>Cerebro_v1.3$addTechnicalInfo()</code></a></p></li>
<li><p><a href='#method-getTechnicalInfo'><code>Cerebro_v1.3$getTechnicalInfo()</code></a></p></li>
<li><p><a href='#method-addGroup'><code>Cerebro_v1.3$addGroup()</code></a></p></li>
<li><p><a href='#method-getGroups'><code>Cerebro_v1.3$getGroups()</code></a></p></li>
<li><p><a href='#method-getGroupLevels'><code>Cerebro_v1.3$getGroupLevels()</code></a></p></li>
<li><p><a href='#method-setMetaData'><code>Cerebro_v1.3$setMetaData()</code></a></p></li>
<li><p><a href='#method-getMetaData'><code>Cerebro_v1.3$getMetaData()</code></a></p></li>
<li><p><a href='#method-addGeneList'><code>Cerebro_v1.3$addGeneList()</code></a></p></li>
<li><p><a href='#method-getGeneLists'><code>Cerebro_v1.3$getGeneLists()</code></a></p></li>
<li><p><a href='#method-setExpression'><code>Cerebro_v1.3$setExpression()</code></a></p></li>
<li><p><a href='#method-getCellNames'><code>Cerebro_v1.3$getCellNames()</code></a></p></li>
<li><p><a href='#method-getGeneNames'><code>Cerebro_v1.3$getGeneNames()</code></a></p></li>
<li><p><a href='#method-getMeanExpressionForGenes'><code>Cerebro_v1.3$getMeanExpressionForGenes()</code></a></p></li>
<li><p><a href='#method-getMeanExpressionForCells'><code>Cerebro_v1.3$getMeanExpressionForCells()</code></a></p></li>
<li><p><a href='#method-getExpressionMatrix'><code>Cerebro_v1.3$getExpressionMatrix()</code></a></p></li>
<li><p><a href='#method-setCellCycle'><code>Cerebro_v1.3$setCellCycle()</code></a></p></li>
<li><p><a href='#method-getCellCycle'><code>Cerebro_v1.3$getCellCycle()</code></a></p></li>
<li><p><a href='#method-addProjection'><code>Cerebro_v1.3$addProjection()</code></a></p></li>
<li><p><a href='#method-availableProjections'><code>Cerebro_v1.3$availableProjections()</code></a></p></li>
<li><p><a href='#method-getProjection'><code>Cerebro_v1.3$getProjection()</code></a></p></li>
<li><p><a href='#method-addTree'><code>Cerebro_v1.3$addTree()</code></a></p></li>
<li><p><a href='#method-getTree'><code>Cerebro_v1.3$getTree()</code></a></p></li>
<li><p><a href='#method-addMostExpressedGenes'><code>Cerebro_v1.3$addMostExpressedGenes()</code></a></p></li>
<li><p><a href='#method-getGroupsWithMostExpressedGenes'><code>Cerebro_v1.3$getGroupsWithMostExpressedGenes()</code></a></p></li>
<li><p><a href='#method-getMostExpressedGenes'><code>Cerebro_v1.3$getMostExpressedGenes()</code></a></p></li>
<li><p><a href='#method-addMarkerGenes'><code>Cerebro_v1.3$addMarkerGenes()</code></a></p></li>
<li><p><a href='#method-getMethodsForMarkerGenes'><code>Cerebro_v1.3$getMethodsForMarkerGenes()</code></a></p></li>
<li><p><a href='#method-getGroupsWithMarkerGenes'><code>Cerebro_v1.3$getGroupsWithMarkerGenes()</code></a></p></li>
<li><p><a href='#method-getMarkerGenes'><code>Cerebro_v1.3$getMarkerGenes()</code></a></p></li>
<li><p><a href='#method-addEnrichedPathways'><code>Cerebro_v1.3$addEnrichedPathways()</code></a></p></li>
<li><p><a href='#method-getMethodsForEnrichedPathways'><code>Cerebro_v1.3$getMethodsForEnrichedPathways()</code></a></p></li>
<li><p><a href='#method-getGroupsWithEnrichedPathways'><code>Cerebro_v1.3$getGroupsWithEnrichedPathways()</code></a></p></li>
<li><p><a href='#method-getEnrichedPathways'><code>Cerebro_v1.3$getEnrichedPathways()</code></a></p></li>
<li><p><a href='#method-addTrajectory'><code>Cerebro_v1.3$addTrajectory()</code></a></p></li>
<li><p><a href='#method-getMethodsForTrajectories'><code>Cerebro_v1.3$getMethodsForTrajectories()</code></a></p></li>
<li><p><a href='#method-getNamesOfTrajectories'><code>Cerebro_v1.3$getNamesOfTrajectories()</code></a></p></li>
<li><p><a href='#method-getTrajectory'><code>Cerebro_v1.3$getTrajectory()</code></a></p></li>
<li><p><a href='#method-addExtraMaterial'><code>Cerebro_v1.3$addExtraMaterial()</code></a></p></li>
<li><p><a href='#method-addExtraTable'><code>Cerebro_v1.3$addExtraTable()</code></a></p></li>
<li><p><a href='#method-addExtraPlot'><code>Cerebro_v1.3$addExtraPlot()</code></a></p></li>
<li><p><a href='#method-getExtraMaterialCategories'><code>Cerebro_v1.3$getExtraMaterialCategories()</code></a></p></li>
<li><p><a href='#method-checkForExtraTables'><code>Cerebro_v1.3$checkForExtraTables()</code></a></p></li>
<li><p><a href='#method-getNamesOfExtraTables'><code>Cerebro_v1.3$getNamesOfExtraTables()</code></a></p></li>
<li><p><a href='#method-getExtraTable'><code>Cerebro_v1.3$getExtraTable()</code></a></p></li>
<li><p><a href='#method-checkForExtraPlots'><code>Cerebro_v1.3$checkForExtraPlots()</code></a></p></li>
<li><p><a href='#method-getNamesOfExtraPlots'><code>Cerebro_v1.3$getNamesOfExtraPlots()</code></a></p></li>
<li><p><a href='#method-getExtraPlot'><code>Cerebro_v1.3$getExtraPlot()</code></a></p></li>
<li><p><a href='#method-print'><code>Cerebro_v1.3$print()</code></a></p></li>
<li><p><a href='#method-clone'><code>Cerebro_v1.3$clone()</code></a></p></li>
</ul>
<p><hr>
<a id="method-new"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>new()</code></h3>
<p>Create a new <code>Cerebro_v1.3</code> object.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>new</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p>A new <code>Cerebro_v1.3</code> object.</p>
<p><hr>
<a id="method-setVersion"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>setVersion()</code></h3>
<p>Set the version of <code>cerebroApp</code> that was used to generate this
object.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>setVersion</span><span class='op'>(</span><span class='va'>version</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>version</code></dt><dd><p>Version to set.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getVersion"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getVersion()</code></h3>
<p>Get the version of <code>cerebroApp</code> that was used to generate this
object.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getVersion</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p>Version as <code>package_version</code> class.</p>
<p><hr>
<a id="method-checkIfGroupExists"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>checkIfGroupExists()</code></h3>
<p>Safety function that will check if a provided group name is present in
the <code>groups</code> field.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>checkIfGroupExists</span><span class='op'>(</span><span class='va'>group_name</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>group_name</code></dt><dd><p>Group name to be tested</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-checkIfColumnExistsInMetadata"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>checkIfColumnExistsInMetadata()</code></h3>
<p>Safety function that will check if a provided group name is present in
the meta data.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>checkIfColumnExistsInMetadata</span><span class='op'>(</span><span class='va'>group_name</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>group_name</code></dt><dd><p>Group name to be tested.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-addExperiment"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addExperiment()</code></h3>
<p>Add information to <code>experiment</code> field.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addExperiment</span><span class='op'>(</span><span class='va'>field</span>, <span class='va'>content</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>field</code></dt><dd><p>Name of the information, e.g. <code>organism</code>.</p></dd>

<dt><code>content</code></dt><dd><p>Actual information, e.g. <code>hg</code>.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getExperiment"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getExperiment()</code></h3>
<p>Retrieve information from <code>experiment</code> field.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getExperiment</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>list</code> of all entries in the <code>experiment</code> field.</p>
<p><hr>
<a id="method-addParameters"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addParameters()</code></h3>
<p>Add information to <code>parameters</code> field.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addParameters</span><span class='op'>(</span><span class='va'>field</span>, <span class='va'>content</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>field</code></dt><dd><p>Name of the information, e.g. <code>number_of_PCs</code>.</p></dd>

<dt><code>content</code></dt><dd><p>Actual information, e.g. <code>30</code>.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getParameters"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getParameters()</code></h3>
<p>Retrieve information from <code>parameters</code> field.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getParameters</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>list</code> of all entries in the <code>parameters</code> field.</p>
<p><hr>
<a id="method-addTechnicalInfo"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addTechnicalInfo()</code></h3>
<p>Add information to <code>technical_info</code> field.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addTechnicalInfo</span><span class='op'>(</span><span class='va'>field</span>, <span class='va'>content</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>field</code></dt><dd><p>Name of the information, e.g. <code>R</code>.</p></dd>

<dt><code>content</code></dt><dd><p>Actual information, e.g. <code>4.0.2</code>.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getTechnicalInfo"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getTechnicalInfo()</code></h3>
<p>Retrieve information from <code>technical_info</code> field.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getTechnicalInfo</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>list</code> of all entries in the <code>technical_info</code> field.</p>
<p><hr>
<a id="method-addGroup"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addGroup()</code></h3>
<p>Add group to the groups registered in the <code>groups</code> field.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addGroup</span><span class='op'>(</span><span class='va'>group_name</span>, <span class='va'>levels</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>group_name</code></dt><dd><p>Group name.</p></dd>

<dt><code>levels</code></dt><dd><p><code>vector</code> of group levels (subgroups).</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getGroups"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getGroups()</code></h3>
<p>Retrieve all names in the <code>groups</code> field.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getGroups</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> of registered groups.</p>
<p><hr>
<a id="method-getGroupLevels"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getGroupLevels()</code></h3>
<p>Retrieve group levels for a group registered in the <code>groups</code> field.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getGroupLevels</span><span class='op'>(</span><span class='va'>group_name</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>group_name</code></dt><dd><p>Group name for which to retrieve group levels.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> of group levels.</p>
<p><hr>
<a id="method-setMetaData"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>setMetaData()</code></h3>
<p>Set meta data for cells.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>setMetaData</span><span class='op'>(</span><span class='va'>table</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>table</code></dt><dd><p><code>data.frame</code> that contains meta data for cells. The
number of rows must be equal to the number of rows of projections and
the number of columns in the transcript count matrix.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getMetaData"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getMetaData()</code></h3>
<p>Retrieve meta data for cells.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getMetaData</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>data.frame</code> containing meta data.</p>
<p><hr>
<a id="method-addGeneList"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addGeneList()</code></h3>
<p>Add a gene list to the <code>gene_lists</code>.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addGeneList</span><span class='op'>(</span><span class='va'>name</span>, <span class='va'>genes</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>name</code></dt><dd><p>Name of the gene list.</p></dd>

<dt><code>genes</code></dt><dd><p><code>vector</code> of genes.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getGeneLists"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getGeneLists()</code></h3>
<p>Retrieve gene lists from the <code>gene_lists</code>.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getGeneLists</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>list</code> of all entries in the <code>gene_lists</code> field.</p>
<p><hr>
<a id="method-setExpression"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>setExpression()</code></h3>
<p>Set transcript count matrix.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>setExpression</span><span class='op'>(</span><span class='va'>counts</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>counts</code></dt><dd><p><code>matrix</code>-like object that contains transcript counts
for cells in the data set. Number of columns must be equal to the number
of rows in the <code>meta_data</code> field.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getCellNames"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getCellNames()</code></h3>
<p>Get names of all cells.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getCellNames</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> containing all cell names/barcodes.</p>
<p><hr>
<a id="method-getGeneNames"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getGeneNames()</code></h3>
<p>Get names of all genes in transcript count matrix.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getGeneNames</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> containing all gene names in transcript count matrix.</p>
<p><hr>
<a id="method-getMeanExpressionForGenes"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getMeanExpressionForGenes()</code></h3>
<p>Retrieve mean expression across all cells in the data set for a set of
genes.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getMeanExpressionForGenes</span><span class='op'>(</span><span class='va'>genes</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>genes</code></dt><dd><p>Names of genes to extract; no default.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>data.frame</code> containing specified gene names and their respective
mean expression across all cells in the data set.</p>
<p><hr>
<a id="method-getMeanExpressionForCells"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getMeanExpressionForCells()</code></h3>
<p>Retrieve (mean) expression for a single gene or a set of genes for a
given set of cells.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getMeanExpressionForCells</span><span class='op'>(</span>cells <span class='op'>=</span> <span class='cn'>NULL</span>, genes <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>cells</code></dt><dd><p>Names/barcodes of cells to extract; defaults to <code>NULL</code>,
which will return all cells.</p></dd>

<dt><code>genes</code></dt><dd><p>Names of genes to extract; defaults to <code>NULL</code>, which
will return all genes.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> containing (mean) expression across all specified genes in
each specified cell.</p>
<p><hr>
<a id="method-getExpressionMatrix"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getExpressionMatrix()</code></h3>
<p>Retrieve transcript count matrix.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getExpressionMatrix</span><span class='op'>(</span>cells <span class='op'>=</span> <span class='cn'>NULL</span>, genes <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>cells</code></dt><dd><p>Names/barcodes of cells to extract; defaults to <code>NULL</code>,
which will return all cells.</p></dd>

<dt><code>genes</code></dt><dd><p>Names of genes to extract; defaults to <code>NULL</code>, which
will return all genes.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p>Dense transcript count matrix for specified cells and genes.</p>
<p><hr>
<a id="method-setCellCycle"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>setCellCycle()</code></h3>
<p>Add columns containing cell cycle assignments to the <code>cell_cycle</code>
field.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>setCellCycle</span><span class='op'>(</span><span class='va'>cols</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>cols</code></dt><dd><p><code>vector</code> of columns names containing cell cycle
assignments.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getCellCycle"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getCellCycle()</code></h3>
<p>Retrieve column names containing cell cycle assignments.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getCellCycle</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> of column names in meta data.</p>
<p><hr>
<a id="method-addProjection"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addProjection()</code></h3>
<p>Add projections (dimensional reductions).</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addProjection</span><span class='op'>(</span><span class='va'>name</span>, <span class='va'>projection</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>name</code></dt><dd><p>Name of the projection.</p></dd>

<dt><code>projection</code></dt><dd><p><code>data.frame</code> containing positions of cells in
projection.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-availableProjections"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>availableProjections()</code></h3>
<p>Get list of available projections (dimensional reductions).</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>availableProjections</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> of projections / dimensional reductions that are available.</p>
<p><hr>
<a id="method-getProjection"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getProjection()</code></h3>
<p>Retrieve data for a specific projection.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getProjection</span><span class='op'>(</span><span class='va'>name</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>name</code></dt><dd><p>Name of projection.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>data.frame</code> containing the positions of cells in the projection.</p>
<p><hr>
<a id="method-addTree"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addTree()</code></h3>
<p>Add phylogenetic tree to <code>trees</code> field.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addTree</span><span class='op'>(</span><span class='va'>group_name</span>, <span class='va'>tree</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>group_name</code></dt><dd><p>Group name that this tree belongs to.</p></dd>

<dt><code>tree</code></dt><dd><p>Phylogenetic tree as <code>phylo</code> object.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getTree"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getTree()</code></h3>
<p>Retrieve phylogenetic tree for a specific group.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getTree</span><span class='op'>(</span><span class='va'>group_name</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>group_name</code></dt><dd><p>Group name for which to retrieve phylogenetic tree.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p>Phylogenetic tree as <code>phylo</code> object.</p>
<p><hr>
<a id="method-addMostExpressedGenes"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addMostExpressedGenes()</code></h3>
<p>Add table of most expressed genes.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addMostExpressedGenes</span><span class='op'>(</span><span class='va'>group_name</span>, <span class='va'>table</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>group_name</code></dt><dd><p>Name of grouping variable that the most expressed genes
belong to. Must be registered in the <code>groups</code> field.</p></dd>

<dt><code>table</code></dt><dd><p><code>data.frame</code> that contains the most expressed genes.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getGroupsWithMostExpressedGenes"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getGroupsWithMostExpressedGenes()</code></h3>
<p>Retrieve names of grouping variables for which most expressed genes are
available.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getGroupsWithMostExpressedGenes</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> of grouping variables for which most expressed genes are
available.</p>
<p><hr>
<a id="method-getMostExpressedGenes"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code><a href='getMostExpressedGenes.html'>getMostExpressedGenes()</a></code></h3>
<p>Retrieve table of most expressed genes for a grouping variable.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getMostExpressedGenes</span><span class='op'>(</span><span class='va'>group_name</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>group_name</code></dt><dd><p>Grouping variable for which most expressed genes should
be retrieved.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>data.frame</code> that contains most expressed genes for group levels of
the specified grouping variable.</p>
<p><hr>
<a id="method-addMarkerGenes"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addMarkerGenes()</code></h3>
<p>Add table of marker genes.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addMarkerGenes</span><span class='op'>(</span><span class='va'>method</span>, <span class='va'>name</span>, <span class='va'>table</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>method</code></dt><dd><p>Name of method that was used to generate the marker genes.</p></dd>

<dt><code>name</code></dt><dd><p>Name of table. This name will be used to select the table in
Cerebro. It is recommended to use the grouping variable, e.g.
<code>sample</code>.</p></dd>

<dt><code>table</code></dt><dd><p><code>data.frame</code> that contains the marker genes.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getMethodsForMarkerGenes"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getMethodsForMarkerGenes()</code></h3>
<p>Retrieve names of methods that were used to generate marker genes.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getMethodsForMarkerGenes</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> of names of methods that were used to generate marker
genes.</p>
<p><hr>
<a id="method-getGroupsWithMarkerGenes"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getGroupsWithMarkerGenes()</code></h3>
<p>Retrieve grouping variables for which marker genes were generated using
a specified method.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getGroupsWithMarkerGenes</span><span class='op'>(</span><span class='va'>method</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>method</code></dt><dd><p>Name of method.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> of grouping variables for which marker genes were
calculated using the specified method.</p>
<p><hr>
<a id="method-getMarkerGenes"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code><a href='getMarkerGenes.html'>getMarkerGenes()</a></code></h3>
<p>Retrieve table of marker genes for specific method and grouping variable.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getMarkerGenes</span><span class='op'>(</span><span class='va'>method</span>, <span class='va'>name</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>method</code></dt><dd><p>Name of method.</p></dd>

<dt><code>name</code></dt><dd><p>Name of table.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>data.frame</code> that contains marker genes for the specified
combination of method and grouping variable.</p>
<p><hr>
<a id="method-addEnrichedPathways"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addEnrichedPathways()</code></h3>
<p>Add table of enriched pathways.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addEnrichedPathways</span><span class='op'>(</span><span class='va'>method</span>, <span class='va'>name</span>, <span class='va'>table</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>method</code></dt><dd><p>Name of method that was used to generate the enriched
pathways.</p></dd>

<dt><code>name</code></dt><dd><p>Name of table. This name will be used to select the table in
Cerebro. It is recommended to use the grouping variable, e.g.
<code>sample</code>.</p></dd>

<dt><code>table</code></dt><dd><p><code>data.frame</code> that contains the enriched pathways.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getMethodsForEnrichedPathways"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getMethodsForEnrichedPathways()</code></h3>
<p>Retrieve names of methods that were used to generate enriched pathways.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getMethodsForEnrichedPathways</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> of names of methods that were used to generate enriched
pathways.</p>
<p><hr>
<a id="method-getGroupsWithEnrichedPathways"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getGroupsWithEnrichedPathways()</code></h3>
<p>Retrieve grouping variables for which enriched pathways were generated
using a specified method.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getGroupsWithEnrichedPathways</span><span class='op'>(</span><span class='va'>method</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>method</code></dt><dd><p>Name of method.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> of grouping variables for which enriched pathways were
calculated using the specified method.</p>
<p><hr>
<a id="method-getEnrichedPathways"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code><a href='getEnrichedPathways.html'>getEnrichedPathways()</a></code></h3>
<p>Retrieve table of enriched pathways for specific method and grouping
variable.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getEnrichedPathways</span><span class='op'>(</span><span class='va'>method</span>, <span class='va'>name</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>method</code></dt><dd><p>Name of method.</p></dd>

<dt><code>name</code></dt><dd><p>Grouping variable.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>data.frame</code> that contains enriched pathways for the specified
combination of method and grouping variable.</p>
<p><hr>
<a id="method-addTrajectory"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addTrajectory()</code></h3>
<p>Add trajectory.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addTrajectory</span><span class='op'>(</span><span class='va'>method</span>, <span class='va'>name</span>, <span class='va'>content</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>method</code></dt><dd><p>Name of method that was used to generate the trajectory.</p></dd>

<dt><code>name</code></dt><dd><p>Name of the trajectory. This name will be used later in
Cerebro to select the trajectory.</p></dd>

<dt><code>content</code></dt><dd><p>Relevant data for the trajectory, depending on the method
this could be a <code>list</code> holding edges, cell positions, pseudotime,
etc.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getMethodsForTrajectories"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getMethodsForTrajectories()</code></h3>
<p>Retrieve names of methods that were used to generate trajectories.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getMethodsForTrajectories</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> of names of methods that were used to generate
trajectories.</p>
<p><hr>
<a id="method-getNamesOfTrajectories"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getNamesOfTrajectories()</code></h3>
<p>Retrieve names of available trajectories for a specified method.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getNamesOfTrajectories</span><span class='op'>(</span><span class='va'>method</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>method</code></dt><dd><p>Name of method.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> of available trajectory for the specified method.</p>
<p><hr>
<a id="method-getTrajectory"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getTrajectory()</code></h3>
<p>Retrieve data for a specific trajectory.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getTrajectory</span><span class='op'>(</span><span class='va'>method</span>, <span class='va'>name</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>method</code></dt><dd><p>Name of method.</p></dd>

<dt><code>name</code></dt><dd><p>Name of trajectory.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p>The type of data depends on the method that was used to generate the
trajectory.</p>
<p><hr>
<a id="method-addExtraMaterial"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addExtraMaterial()</code></h3>
<p>Add content to extra material field.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addExtraMaterial</span><span class='op'>(</span><span class='va'>category</span>, <span class='va'>name</span>, <span class='va'>content</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>category</code></dt><dd><p>Name of category. At the moment, only <code>tables</code> and
<code>plots</code> are valid categories. Tables must be in <code>data.frame</code>
format and plots must be created with <code>ggplot2</code>.</p></dd>

<dt><code>name</code></dt><dd><p>Name of material, will be used to select it in Cerebro.</p></dd>

<dt><code>content</code></dt><dd><p>Data that should be added.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-addExtraTable"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addExtraTable()</code></h3>
<p>Add table to `extra_material` slot.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addExtraTable</span><span class='op'>(</span><span class='va'>name</span>, <span class='va'>table</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>name</code></dt><dd><p>Name of material, will be used to select it in Cerebro.</p></dd>

<dt><code>table</code></dt><dd><p>Table that should be added, must be <code>data.frame</code>.</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-addExtraPlot"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>addExtraPlot()</code></h3>
<p>Add plot to `extra_material` slot.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>addExtraPlot</span><span class='op'>(</span><span class='va'>name</span>, <span class='va'>plot</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>name</code></dt><dd><p>Name of material, will be used to select it in Cerebro.</p></dd>

<dt><code>plot</code></dt><dd><p>Plot that should be added, must be created with
<code>ggplot2</code> (class: <code>ggplot</code>).</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-getExtraMaterialCategories"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getExtraMaterialCategories()</code></h3>
<p>Get names of categories for which extra material is available.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getExtraMaterialCategories</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> with names of available categories.</p>
<p><hr>
<a id="method-checkForExtraTables"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>checkForExtraTables()</code></h3>
<p>Check whether there are tables in the extra materials.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>checkForExtraTables</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>logical</code> indicating whether there are tables in the extra
materials.</p>
<p><hr>
<a id="method-getNamesOfExtraTables"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getNamesOfExtraTables()</code></h3>
<p>Get names of tables in extra materials.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getNamesOfExtraTables</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> containing names of tables in extra materials.</p>
<p><hr>
<a id="method-getExtraTable"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getExtraTable()</code></h3>
<p>Get table from extra materials.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getExtraTable</span><span class='op'>(</span><span class='va'>name</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>name</code></dt><dd><p>Name of table.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p>Requested table in <code>data.frame</code> format.</p>
<p><hr>
<a id="method-checkForExtraPlots"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>checkForExtraPlots()</code></h3>
<p>Check whether there are plots in the extra materials.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>checkForExtraPlots</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>logical</code> indicating whether there are plots in the extra
materials.</p>
<p><hr>
<a id="method-getNamesOfExtraPlots"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getNamesOfExtraPlots()</code></h3>
<p>Get names of plots in extra materials.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getNamesOfExtraPlots</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p><code>vector</code> containing names of plots in extra materials.</p>
<p><hr>
<a id="method-getExtraPlot"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>getExtraPlot()</code></h3>
<p>Get plot from extra materials.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>getExtraPlot</span><span class='op'>(</span><span class='va'>name</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>name</code></dt><dd><p>Name of plot.</p></dd>

</dl><p></div></p>
<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Returns</h4>
<p>Requested plot made with <code>ggplot2</code>.</p>
<p><hr>
<a id="method-print"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code><a href='https://rdrr.io/r/base/print.html'>print()</a></code></h3>
<p>Show overview of object and the data it contains.
Print overview of available marker gene results for <code>self$print()</code>
function.
Print overview of available enriched pathway results for
<code>self$print()</code> function.
Print overview of available trajectories for <code>self$print()</code> function.
Print overview of extra material for <code>self$print()</code> function.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>print</span><span class='op'>(</span><span class='op'>)</span></pre><p></div></p>

<p><hr>
<a id="method-clone"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>clone()</code></h3>
<p>The objects of this class are cloneable with this method.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>Cerebro_v1.3</span><span class='op'>$</span><span class='fu'>clone</span><span class='op'>(</span>deep <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>deep</code></dt><dd><p>Whether to make a deep clone.</p></dd>

</dl><p></div></p>



  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Roman Hillje.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


