<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Export SingleCellExperiment (SCE) object to Cerebro. — exportFromSCE • cerebroApp</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Export SingleCellExperiment (SCE) object to Cerebro. — exportFromSCE" />
<meta property="og:description" content="This function allows to export a SingleCellExperiment (SCE)
object to visualize in Cerebro." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cerebroApp</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/cerebroApp_workflow_Seurat.html">The cerebroApp workflow (Seurat)</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">The Cerebro interface</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/Cerebro_interface_v1.2.html">Cerebro v1.2 and older</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="../articles/export_a_data_set_in_SCE_format.html">Export an SCE/SingleCellExperiment object</a>
    </li>
    <li>
      <a href="../articles/export_and_visualize_custom_tables_and_plots.html">Export and visualize custom tables and plots</a>
    </li>
    <li>
      <a href="../articles/launch_cerebro_with_pre-loaded_data_set.html">Launch Cerebro with pre-loaded data set</a>
    </li>
    <li>
      <a href="../articles/host_cerebro_on_shinyapps.html">Host Cerebro on shinyapps.io</a>
    </li>
    <li>
      <a href="../articles/control_access_to_cerebro_with_a_login_page.html">Control access to Cerebro with a login page</a>
    </li>
    <li>
      <a href="../articles/overview_of_cerebro_v1.3_class.html">Overview of the Cerebro v1.3 class</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Release notes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/release_notes_v1.3.html">v1.3</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/romanhaa/cerebroApp/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/fakechek1">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Export SingleCellExperiment (SCE) object to Cerebro.</h1>
    <small class="dont-index">Source: <a href='https://github.com/romanhaa/cerebroApp/blob/master/R/exportFromSCE.R'><code>R/exportFromSCE.R</code></a></small>
    <div class="hidden name"><code>exportFromSCE.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function allows to export a <code>SingleCellExperiment</code> (<code>SCE</code>)
object to visualize in Cerebro.</p>
    </div>

    <pre class="usage"><span class='fu'>exportFromSCE</span><span class='op'>(</span>
  <span class='va'>object</span>,
  assay <span class='op'>=</span> <span class='st'>"logcounts"</span>,
  <span class='va'>file</span>,
  <span class='va'>experiment_name</span>,
  <span class='va'>organism</span>,
  <span class='va'>groups</span>,
  cell_cycle <span class='op'>=</span> <span class='cn'>NULL</span>,
  nUMI <span class='op'>=</span> <span class='st'>"nUMI"</span>,
  nGene <span class='op'>=</span> <span class='st'>"nGene"</span>,
  add_all_meta_data <span class='op'>=</span> <span class='cn'>TRUE</span>,
  use_delayed_array <span class='op'>=</span> <span class='cn'>FALSE</span>,
  verbose <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p><code>SingleCellExperiment</code> (<code>SCE</code>) object.</p></td>
    </tr>
    <tr>
      <th>assay</th>
      <td><p>Assay to pull expression values from; defaults to
<code>logcounts</code>. It is recommended to use sparse data (such as
log-transformed or raw counts) instead of dense data (such as the 'scaled'
slot) to avoid performance bottlenecks in the Cerebro interface.</p></td>
    </tr>
    <tr>
      <th>file</th>
      <td><p>Where to save the output.</p></td>
    </tr>
    <tr>
      <th>experiment_name</th>
      <td><p>Experiment name.</p></td>
    </tr>
    <tr>
      <th>organism</th>
      <td><p>Organism, e.g. <code>hg</code> (human), <code>mm</code> (mouse), etc.</p></td>
    </tr>
    <tr>
      <th>groups</th>
      <td><p>Names of grouping variables in meta data
(<code>colData(object)</code>), e.g. <code><a href='https://rdrr.io/r/base/c.html'>c("sample","cluster")</a></code>; at least one
must be provided; defaults to <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>cell_cycle</th>
      <td><p>Names of columns in meta data (<code>colData(object)</code>) that#
contain cell cycle information, e.g. <code><a href='https://rdrr.io/r/base/c.html'>c("Phase")</a></code>; defaults to
<code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>nUMI</th>
      <td><p>Column in <code>colData(object)</code> that contains information about
number of transcripts per cell; defaults to <code>nUMI</code>.</p></td>
    </tr>
    <tr>
      <th>nGene</th>
      <td><p>Column in <code>colData(object)</code> that contains information about
number of expressed genes per cell; defaults to <code>nGene</code>.</p></td>
    </tr>
    <tr>
      <th>add_all_meta_data</th>
      <td><p>If set to <code>TRUE</code>, all further meta data columns
will be extracted as well.</p></td>
    </tr>
    <tr>
      <th>use_delayed_array</th>
      <td><p>When set to <code>TRUE</code>, the expression matrix will
be stored as an <code>RleMatrix</code> (see <code>DelayedArray</code> package). This can
be useful for very large data sets, as the matrix won't be loaded into memory
and instead values will be read from the disk directly, at the cost of
performance. Note that it is necessary to install the <code>DelayedArray</code>
package. If set to <code>FALSE</code> (default), the expression matrix will be
copied from the input object as is. It is recommended to use a sparse format,
such as <code>dgCMatrix</code> from the <code>Matrix</code> package.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Set this to <code>TRUE</code> if you want additional log messages;
defaults to <code>FALSE</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>No data returned.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>pbmc</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata/v1.3/pbmc_SCE.rds"</span>,
  package <span class='op'>=</span> <span class='st'>"cerebroApp"</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'>exportFromSCE</span><span class='op'>(</span>
  object <span class='op'>=</span> <span class='va'>pbmc</span>,
  file <span class='op'>=</span> <span class='st'>'pbmc_SCE.crb'</span>,
  experiment_name <span class='op'>=</span> <span class='st'>'PBMC'</span>,
  organism <span class='op'>=</span> <span class='st'>'hg'</span>,
  groups <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'sample'</span>,<span class='st'>'cluster'</span><span class='op'>)</span>,
  nUMI <span class='op'>=</span> <span class='st'>'nUMI'</span>,
  nGene <span class='op'>=</span> <span class='st'>'nGene'</span>,
  use_delayed_array <span class='op'>=</span> <span class='cn'>FALSE</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Loading required package: SingleCellExperiment</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: SummarizedExperiment</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: MatrixGenerics</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: matrixStats</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘MatrixGenerics’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:matrixStats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span>
#&gt; <span class='message'>    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span>
#&gt; <span class='message'>    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span>
#&gt; <span class='message'>    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span>
#&gt; <span class='message'>    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span>
#&gt; <span class='message'>    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span>
#&gt; <span class='message'>    colWeightedMeans, colWeightedMedians, colWeightedSds,</span>
#&gt; <span class='message'>    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span>
#&gt; <span class='message'>    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span>
#&gt; <span class='message'>    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span>
#&gt; <span class='message'>    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span>
#&gt; <span class='message'>    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span>
#&gt; <span class='message'>    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span>
#&gt; <span class='message'>    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span>
#&gt; <span class='message'>    rowWeightedSds, rowWeightedVars</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomicRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: stats4</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocGenerics</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: parallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘BiocGenerics’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:parallel’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
#&gt; <span class='message'>    clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
#&gt; <span class='message'>    parLapplyLB, parRapply, parSapply, parSapplyLB</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
#&gt; <span class='message'>    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
#&gt; <span class='message'>    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
#&gt; <span class='message'>    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
#&gt; <span class='message'>    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
#&gt; <span class='message'>    union, unique, unsplit, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: S4Vectors</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘S4Vectors’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: IRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomeInfoDb</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: Biobase</span></div><div class='output co'>#&gt; <span class='message'>Welcome to Bioconductor</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    Vignettes contain introductory material; view with</span>
#&gt; <span class='message'>    'browseVignettes()'. To cite Bioconductor, see</span>
#&gt; <span class='message'>    'citation("Biobase")', and for packages 'citation("pkgname")'.</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘Biobase’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:MatrixGenerics’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    rowMedians</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:matrixStats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyMissing, rowMedians</span></div><div class='output co'>#&gt; <span class='message'>[17:53:23] Initializing Cerebro object...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: the condition has length &gt; 1 and only the first element will be used</span></div><div class='output co'>#&gt; <span class='message'>[17:53:23] Collecting available meta data...</span></div><div class='output co'>#&gt; <span class='message'>[17:53:23] Extracting all meta data columns...</span></div><div class='output co'>#&gt; <span class='message'>[17:53:23] Extracting dimensional reductions...</span></div><div class='output co'>#&gt; <span class='message'>[17:53:23] Will export the following dimensional reductions: UMAP</span></div><div class='output co'>#&gt; <span class='message'>[17:53:23] No trajectories to extract...</span></div><div class='output co'>#&gt; <span class='message'>[17:53:23] Overview of Cerebro object:</span></div><div class='output co'>#&gt; <span class='message'>class: Cerebro_v1.3</span>
#&gt; <span class='message'>cerebroApp version: 1.3.1</span>
#&gt; <span class='message'>experiment name: PBMC</span>
#&gt; <span class='message'>organism: hg</span>
#&gt; <span class='message'>date of analysis: </span>
#&gt; <span class='message'>date of export: 2021-03-12</span>
#&gt; <span class='message'>number of cells: 80</span>
#&gt; <span class='message'>number of genes: 230</span>
#&gt; <span class='message'>grouping variables (2): sample, cluster</span>
#&gt; <span class='message'>cell cycle variables (0): </span>
#&gt; <span class='message'>projections (1): UMAP</span>
#&gt; <span class='message'>trees (0): </span>
#&gt; <span class='message'>most expressed genes: </span>
#&gt; <span class='message'>marker genes:</span>
#&gt; <span class='message'>enriched pathways:</span>
#&gt; <span class='message'>trajectories:</span>
#&gt; <span class='message'>extra material:</span></div><div class='output co'>#&gt; <span class='message'>[17:53:23] Saving Cerebro object to: pbmc_SCE.crb</span></div><div class='output co'>#&gt; <span class='message'>[17:53:23] Done!</span></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Roman Hillje.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


