<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Perform gene set enrichment analysis with GSVA. — performGeneSetEnrichmentAnalysis • cerebroApp</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Perform gene set enrichment analysis with GSVA. — performGeneSetEnrichmentAnalysis" />
<meta property="og:description" content="This function calculates enrichment scores, p- and q-value statistics for
provided gene sets for specified groups of cells in given Seurat object using
gene set variation analysis (GSVA). Calculation of p- and q-values for gene
sets is performed as done in &quot;Evaluation of methods to assign cell type
labels to cell clusters from single-cell RNA-sequencing data&quot;, Diaz-Mejia et
al., F1000Research (2019)." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cerebroApp</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/cerebroApp_workflow_Seurat.html">The cerebroApp workflow (Seurat)</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">The Cerebro interface</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="../articles/Cerebro_interface_v1.2.html">Cerebro v1.2 and older</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="../articles/export_a_data_set_in_SCE_format.html">Export an SCE/SingleCellExperiment object</a>
    </li>
    <li>
      <a href="../articles/export_and_visualize_custom_tables_and_plots.html">Export and visualize custom tables and plots</a>
    </li>
    <li>
      <a href="../articles/launch_cerebro_with_pre-loaded_data_set.html">Launch Cerebro with pre-loaded data set</a>
    </li>
    <li>
      <a href="../articles/host_cerebro_on_shinyapps.html">Host Cerebro on shinyapps.io</a>
    </li>
    <li>
      <a href="../articles/control_access_to_cerebro_with_a_login_page.html">Control access to Cerebro with a login page</a>
    </li>
    <li>
      <a href="../articles/overview_of_cerebro_v1.3_class.html">Overview of the Cerebro v1.3 class</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Release notes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/release_notes_v1.3.html">v1.3</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/romanhaa/cerebroApp/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/fakechek1">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Perform gene set enrichment analysis with GSVA.</h1>
    <small class="dont-index">Source: <a href='https://github.com/romanhaa/cerebroApp/blob/master/R/performGeneSetEnrichmentAnalysis.R'><code>R/performGeneSetEnrichmentAnalysis.R</code></a></small>
    <div class="hidden name"><code>performGeneSetEnrichmentAnalysis.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function calculates enrichment scores, p- and q-value statistics for
provided gene sets for specified groups of cells in given Seurat object using
gene set variation analysis (GSVA). Calculation of p- and q-values for gene
sets is performed as done in "Evaluation of methods to assign cell type
labels to cell clusters from single-cell RNA-sequencing data", Diaz-Mejia et
al., F1000Research (2019).</p>
    </div>

    <pre class="usage"><span class='fu'>performGeneSetEnrichmentAnalysis</span><span class='op'>(</span>
  <span class='va'>object</span>,
  assay <span class='op'>=</span> <span class='st'>"RNA"</span>,
  <span class='va'>GMT_file</span>,
  groups <span class='op'>=</span> <span class='cn'>NULL</span>,
  name <span class='op'>=</span> <span class='st'>"cerebro_GSVA"</span>,
  thresh_p_val <span class='op'>=</span> <span class='fl'>0.05</span>,
  thresh_q_val <span class='op'>=</span> <span class='fl'>0.1</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p>Seurat object.</p></td>
    </tr>
    <tr>
      <th>assay</th>
      <td><p>Assay to pull counts from; defaults to 'RNA'. Only relevant in
Seurat v3.0 or higher since the concept of assays wasn't implemented before.</p></td>
    </tr>
    <tr>
      <th>GMT_file</th>
      <td><p>Path to GMT file containing the gene sets to be tested.
The Broad Institute provides many gene sets which can be downloaded:
http://software.broadinstitute.org/gsea/msigdb/index.jsp</p></td>
    </tr>
    <tr>
      <th>groups</th>
      <td><p>Grouping variables (columns) in object@meta.data for which
gene set enrichment analysis should be performed</p></td>
    </tr>
    <tr>
      <th>name</th>
      <td><p>Name of list that should be used to store the results in
object@misc$enriched_pathways$&lt;name&gt;; defaults to 'cerebro_GSVA'.</p></td>
    </tr>
    <tr>
      <th>thresh_p_val</th>
      <td><p>Threshold for p-value, defaults to 0.05.</p></td>
    </tr>
    <tr>
      <th>thresh_q_val</th>
      <td><p>Threshold for q-value, defaults to 0.1.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Further parameters can be passed to control GSVA::gsva().</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Seurat object with GSVA results for the specified grouping variables
stored in object@misc$enriched_pathways$&lt;name&gt;</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>pbmc</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata/v1.3/pbmc_seurat.rds"</span>,
  package <span class='op'>=</span> <span class='st'>"cerebroApp"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>example_gene_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata/example_gene_set.gmt"</span>,
  package <span class='op'>=</span> <span class='st'>"cerebroApp"</span><span class='op'>)</span>
<span class='va'>pbmc</span> <span class='op'>&lt;-</span> <span class='fu'>performGeneSetEnrichmentAnalysis</span><span class='op'>(</span>
  object <span class='op'>=</span> <span class='va'>pbmc</span>,
  GMT_file <span class='op'>=</span> <span class='va'>example_gene_set</span>,
  groups <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'sample'</span>,<span class='st'>'seurat_clusters'</span><span class='op'>)</span>,
  thresh_p_val <span class='op'>=</span> <span class='fl'>0.05</span>,
  thresh_q_val <span class='op'>=</span> <span class='fl'>0.1</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>[17:53:55] Loading gene sets...</span></div><div class='output co'>#&gt; <span class='message'>[17:53:55] Loaded 2 gene sets from GMT file.</span></div><div class='output co'>#&gt; <span class='message'>[17:53:55] Extracting transcript counts from `data` slot of `RNA` assay...</span></div><div class='output co'>#&gt; <span class='message'>[17:53:55] Performing analysis for 2 subgroups of group `sample`...</span></div><div class='output co'>#&gt; Estimating GSVA scores for 2 gene sets.
#&gt; Estimating ECDFs with Gaussian kernels
#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |===================================                                   |  50%  |                                                                              |======================================================================| 100%
#&gt; </div><div class='output co'>#&gt; <span class='message'>[17:53:55] 0 gene sets passed the thresholds across all subgroups of group `sample`.</span></div><div class='output co'>#&gt; <span class='message'>[17:53:55] Performing analysis for 3 subgroups of group `seurat_clusters`...</span></div><div class='output co'>#&gt; Estimating GSVA scores for 2 gene sets.
#&gt; Estimating ECDFs with Gaussian kernels
#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |===================================                                   |  50%  |                                                                              |======================================================================| 100%
#&gt; </div><div class='output co'>#&gt; <span class='message'>[17:53:55] 0 gene sets passed the thresholds across all subgroups of group `seurat_clusters`.</span></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Roman Hillje.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


