<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to the cerebroApp workflow (Seurat) • cerebroApp</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to the cerebroApp workflow (Seurat)">
<meta property="og:description" content="cerebroApp">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cerebroApp</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cerebroApp_workflow_Seurat.html">The cerebroApp workflow (Seurat)</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">The Cerebro interface</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/Cerebro_interface_v1.2.html">Cerebro v1.2 and older</a>
        </li>
      </ul>
</li>
    <li>
      <a href="../articles/export_a_data_set_in_SCE_format.html">Export an SCE/SingleCellExperiment object</a>
    </li>
    <li>
      <a href="../articles/export_and_visualize_custom_tables_and_plots.html">Export and visualize custom tables and plots</a>
    </li>
    <li>
      <a href="../articles/launch_cerebro_with_pre-loaded_data_set.html">Launch Cerebro with pre-loaded data set</a>
    </li>
    <li>
      <a href="../articles/host_cerebro_on_shinyapps.html">Host Cerebro on shinyapps.io</a>
    </li>
    <li>
      <a href="../articles/control_access_to_cerebro_with_a_login_page.html">Control access to Cerebro with a login page</a>
    </li>
    <li>
      <a href="../articles/overview_of_cerebro_v1.3_class.html">Overview of the Cerebro v1.3 class</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Release notes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/release_notes_v1.3.html">v1.3</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/romanhaa/cerebroApp/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/fakechek1">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="cerebroApp_workflow_Seurat_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to the cerebroApp workflow (Seurat)</h1>
                        <h4 class="author">Roman Hillje</h4>
            
            <h4 class="date">12 March, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/romanhaa/cerebroApp/blob/master/vignettes/cerebroApp_workflow_Seurat.Rmd"><code>vignettes/cerebroApp_workflow_Seurat.Rmd</code></a></small>
      <div class="hidden name"><code>cerebroApp_workflow_Seurat.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>The <code>cerebroApp</code> package has two main purposes: (1) Give access to the Cerebro user interface, and (2) provide a set of functions to pre-process and export scRNA-seq data for visualization in Cerebro. The Cerebro user interface was built using the Shiny framework and designed to provide numerous perspectives on a given data set that should facilitate data interpretation. This vignette will give you an example of how scRNA-seq data can be processed using the <a href="https://satijalab.org/seurat/">Seurat</a> toolkit and then exported for visualization in Cerebro.</p>
<p>Ket features that can be accessed through Cerebro:</p>
<ul>
<li>Interactive 2D and 3D dimensional reductions.</li>
<li>Overview panels for all grouping variables, e.g. samples and clusters.</li>
<li>Tables of most expressed genes and marker genes for subgroups of grouping variables.</li>
<li>Tables of enriched pathways for subgroups of grouping variables.</li>
<li>Query gene(s) and gene sets from MSigDB and show their expression in dimensional reductions.</li>
<li>Visualize trajectories calculated with Monocle v2.</li>
<li>All plots can be exported to PNG. In addition, 2D dimensional reductions can be exported to PDF.</li>
<li>Tables can be downloaded in CSV or Excel format.</li>
</ul>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>Before starting the workflow, we need to install <code>cerebroApp</code>, as well as the <a href="https://satijalab.org/seurat/"><code>Seurat</code></a>, <a href="https://cole-trapnell-lab.github.io/monocle-release/"><code>monocle</code></a> and <a href="https://www.bioconductor.org/packages/release/bioc/html/SingleR.html"><code>SingleR</code></a> packages, which are not installed as dependencies of cerebroApp because they are only necessary if you want/need to pre-process your scRNA-seq data. If you just want to launch the Cerebro user interface, e.g. because you already have the pre-processed data, you don’t need those packages.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span> <span class="st">'BiocManager'</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="cn">FALSE</span> <span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">'BiocManager'</span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BiocManager</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span> <span class="st">'cerebroApp'</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="cn">FALSE</span> <span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">'romanhaa/cerebroApp'</span><span class="op">)</span>
<span class="kw">if</span> <span class="op">(</span> <span class="st">'Seurat'</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="cn">FALSE</span> <span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">'Seurat'</span><span class="op">)</span>
<span class="kw">if</span> <span class="op">(</span> <span class="st">'SingleR'</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="cn">FALSE</span> <span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">'SingleR'</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; The downloaded binary packages are in</span>
<span class="co">#&gt;  /var/folders/7z/m3l07m8s6455y1mnykf307_c0000gn/T//RtmpfbO0gJ/downloaded_packages</span>
<span class="kw">if</span> <span class="op">(</span> <span class="st">'monocle'</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="cn">FALSE</span> <span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">'monocle'</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; The downloaded binary packages are in</span>
<span class="co">#&gt;  /var/folders/7z/m3l07m8s6455y1mnykf307_c0000gn/T//RtmpfbO0gJ/downloaded_packages</span></code></pre></div>
</div>
<div id="setup" class="section level1">
<h1 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h1>
<p>Apart from the packages we just installed, we will also load the dplyr package and set a seed to make our analysis reproducible.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/SingleR">SingleR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">monocle</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://romanhaa.github.io/cerebroApp/index.html">cerebroApp</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1234567</span><span class="op">)</span></code></pre></div>
</div>
<div id="load-count-data" class="section level1">
<h1 class="hasAnchor">
<a href="#load-count-data" class="anchor"></a>Load count data</h1>
<p>In this example workflow, we will load a small transcript count table from the <a href="https://satijalab.org/seurat/">Seurat</a> package containing 80 cells and 230 genes. If you want to try the workflow with a real data set, you can find some transcript count matrices on the <a href="https://github.com/romanhaa/Cerebro/tree/master/examples">Cerebro GitHub repo</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>
  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, <span class="st">'pbmc_raw.txt'</span>, package <span class="op">=</span> <span class="st">'Seurat'</span><span class="op">)</span>,
  as.is <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
</div>
<div id="pre-processing-with-seurat" class="section level1">
<h1 class="hasAnchor">
<a href="#pre-processing-with-seurat" class="anchor"></a>Pre-processing with Seurat</h1>
<p>Now, we create a <code>Seurat</code> object and filter out cells with less than <code>50</code> transcripts or fewer than <code>10</code> expressed genes. Those cut-offs are only reasonable for this example data set and will likely need to be adjusted in a real data set. Then, we follow the <a href="https://satijalab.org/seurat/essential_commands.html">standard Seurat workflow</a>, including…</p>
<ul>
<li>log-normalization of transcript counts,</li>
<li>identification of highly variably genes,</li>
<li>scaling of expression values and regression for the number of transcripts per cell,</li>
<li>principal component analysis (PCA).</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/CreateSeuratObject.html">CreateSeuratObject</a></span><span class="op">(</span>
  project <span class="op">=</span> <span class="st">'pbmc'</span>,
  counts <span class="op">=</span> <span class="va">pbmc</span>,
  min.cells <span class="op">=</span> <span class="fl">5</span>
<span class="op">)</span>
<span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/subset.Seurat.html">subset</a></span><span class="op">(</span><span class="va">seurat</span>, subset <span class="op">=</span> <span class="va">nCount_RNA</span> <span class="op">&gt;=</span> <span class="fl">50</span> <span class="op">&amp;</span> <span class="va">nFeature_RNA</span> <span class="op">&gt;=</span> <span class="fl">10</span><span class="op">)</span>
<span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html">NormalizeData</a></span><span class="op">(</span><span class="va">seurat</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html">FindVariableFeatures</a></span><span class="op">(</span><span class="va">seurat</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html">ScaleData</a></span><span class="op">(</span><span class="va">seurat</span>, vars.to.regress <span class="op">=</span> <span class="st">'nCount_RNA'</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html">RunPCA</a></span><span class="op">(</span><span class="va">seurat</span>, features <span class="op">=</span> <span class="va">seurat</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">var.features</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
</div>
<div id="clustering" class="section level1">
<h1 class="hasAnchor">
<a href="#clustering" class="anchor"></a>Clustering</h1>
<p>Then, we identify clusters with a relatively low resolution, again, because of the small size of this data set.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html">FindNeighbors</a></span><span class="op">(</span><span class="va">seurat</span><span class="op">)</span>
<span class="co">#&gt; Computing nearest neighbor graph</span>
<span class="co">#&gt; Computing SNN</span>
<span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html">FindClusters</a></span><span class="op">(</span><span class="va">seurat</span>, resolution <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Number of nodes: 78</span>
<span class="co">#&gt; Number of edges: 2373</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Running Louvain algorithm...</span>
<span class="co">#&gt; Maximum modularity in 10 random starts: 0.5759</span>
<span class="co">#&gt; Number of communities: 2</span>
<span class="co">#&gt; Elapsed time: 0 seconds</span>
<span class="va">seurat</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">RNA_snn_res.0.5</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></code></pre></div>
</div>
<div id="cell-cycle-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#cell-cycle-analysis" class="anchor"></a>Cell cycle analysis</h1>
<p>We could also perform cell cycle analysis using the <code><a href="https://satijalab.org/seurat/reference/CellCycleScoring.html">CellCycleScoring()</a></code> built into Seurat. However, too few of the cell cycle genes are present in this example data set so we will skip it here. The code below should work with a real data set.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/CellCycleScoring.html">CellCycleScoring</a></span><span class="op">(</span>
  <span class="va">seurat</span>,
  g2m.features <span class="op">=</span> <span class="va">cc.genes</span><span class="op">$</span><span class="va">g2m.genes</span>,
  s.features <span class="op">=</span> <span class="va">cc.genes</span><span class="op">$</span><span class="va">s.genes</span>
<span class="op">)</span>

<span class="va">seurat</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">gene_lists</span><span class="op">$</span><span class="va">G2M_phase_genes</span> <span class="op">&lt;-</span> <span class="va">cc.genes</span><span class="op">$</span><span class="va">g2m.genes</span>
<span class="va">seurat</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">gene_lists</span><span class="op">$</span><span class="va">S_phase_genes</span> <span class="op">&lt;-</span> <span class="va">cc.genes</span><span class="op">$</span><span class="va">s.genes</span></code></pre></div>
</div>
<div id="dimensional-reduction" class="section level1">
<h1 class="hasAnchor">
<a href="#dimensional-reduction" class="anchor"></a>Dimensional reduction</h1>
<p>Next, we use the UMAP technique to generate two- and three-dimensional projections.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html">RunUMAP</a></span><span class="op">(</span>
  <span class="va">seurat</span>,
  reduction.name <span class="op">=</span> <span class="st">'UMAP'</span>,
  reduction.key <span class="op">=</span> <span class="st">'UMAP_'</span>,
  dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>,
  n.components <span class="op">=</span> <span class="fl">2</span>,
  seed.use <span class="op">=</span> <span class="fl">100</span>,
  verbose <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span>

<span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html">RunUMAP</a></span><span class="op">(</span>
  <span class="va">seurat</span>,
  reduction.name <span class="op">=</span> <span class="st">'UMAP_3D'</span>,
  reduction.key <span class="op">=</span> <span class="st">'UMAP3D_'</span>,
  dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>,
  n.components <span class="op">=</span> <span class="fl">3</span>,
  seed.use <span class="op">=</span> <span class="fl">100</span>,
  verbose <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
</div>
<div id="curate-meta-data" class="section level1">
<h1 class="hasAnchor">
<a href="#curate-meta-data" class="anchor"></a>Curate meta data</h1>
<p>This example data set consists of a single sample so we just add that name to the meta data. Moreover, in order to be able to understand how we did the analysis later, we add some meta data to the <code>misc</code> slot of our <code>Seurat</code> object, e.g. an experiment name and the species. The info we store there will later be collected for visualization in Cerebro.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">sample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="st">'pbmc'</span>, levels <span class="op">=</span> <span class="st">'pbmc'</span><span class="op">)</span>

<span class="va">seurat</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">experiment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  experiment_name <span class="op">=</span> <span class="st">'pbmc'</span>,
  organism <span class="op">=</span> <span class="st">'hg'</span>,
  date_of_analysis <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html">Sys.Date</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span>

<span class="va">seurat</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">parameters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  gene_nomenclature <span class="op">=</span> <span class="st">'gene_name'</span>,
  discard_genes_expressed_in_fewer_cells_than <span class="op">=</span> <span class="fl">10</span>,
  keep_mitochondrial_genes <span class="op">=</span> <span class="cn">TRUE</span>,
  variables_to_regress_out <span class="op">=</span> <span class="st">'nUMI'</span>,
  number_PCs <span class="op">=</span> <span class="fl">30</span>,
  cluster_resolution <span class="op">=</span> <span class="fl">0.5</span>
<span class="op">)</span>

<span class="va">seurat</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">parameters</span><span class="op">$</span><span class="va">filtering</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  UMI_min <span class="op">=</span> <span class="fl">50</span>,
  UMI_max <span class="op">=</span> <span class="cn">Inf</span>,
  genes_min <span class="op">=</span> <span class="fl">10</span>,
  genes_max <span class="op">=</span> <span class="cn">Inf</span>
<span class="op">)</span>

<span class="va">seurat</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">technical_info</span><span class="op">$</span><span class="va">cerebroApp_version</span> <span class="op">&lt;-</span> <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html">packageVersion</a></span><span class="op">(</span><span class="st">'cerebroApp'</span><span class="op">)</span>
<span class="va">seurat</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">technical_info</span><span class="op">$</span><span class="va">Seurat</span> <span class="op">&lt;-</span> <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html">packageVersion</a></span><span class="op">(</span><span class="st">'Seurat'</span><span class="op">)</span>
<span class="va">seurat</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">technical_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  <span class="st">'R'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/capture.output.html">capture.output</a></span><span class="op">(</span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/sessioninfo/man/session_info.html">session_info</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="co">#&gt; Registered S3 method overwritten by 'cli':</span>
<span class="co">#&gt;   method     from    </span>
<span class="co">#&gt;   print.boxx spatstat</span></code></pre></div>
</div>
<div id="assign-cell-types" class="section level1">
<h1 class="hasAnchor">
<a href="#assign-cell-types" class="anchor"></a>Assign cell types</h1>
<p>Using the <a href="https://www.bioconductor.org/packages/release/bioc/html/SingleR.html"><code>SingleR</code></a> package, we can get a suggestion of cell type for each cell. Apart from the suggested cell type, we also extract the score that might help us to understand how confident the assignment was.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">singler_ref</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleR/man/datasets.html">BlueprintEncodeData</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Warning: 'BlueprintEncodeData' is deprecated.</span>
<span class="co">#&gt; Use 'celldex::BlueprintEncodeData' instead.</span>
<span class="co">#&gt; See help("Deprecated")</span>
<span class="co">#&gt; using temporary cache /var/folders/7z/m3l07m8s6455y1mnykf307_c0000gn/T//RtmpfbO0gJ/BiocFileCache</span>
<span class="co">#&gt; snapshotDate(): 2020-10-27</span>
<span class="co">#&gt; see ?celldex and browseVignettes('celldex') for documentation</span>
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span>
<span class="co">#&gt; see ?celldex and browseVignettes('celldex') for documentation</span>
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span>

<span class="va">singler_results_blueprintencode_main</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleR/man/SingleR.html">SingleR</a></span><span class="op">(</span>
  test <span class="op">=</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/GetAssayData.html">GetAssayData</a></span><span class="op">(</span><span class="va">seurat</span>, assay <span class="op">=</span> <span class="st">'RNA'</span>, slot <span class="op">=</span> <span class="st">'data'</span><span class="op">)</span>,
  ref <span class="op">=</span> <span class="va">singler_ref</span>,
  labels <span class="op">=</span> <span class="va">singler_ref</span><span class="op">@</span><span class="va">colData</span><span class="op">@</span><span class="va">listData</span><span class="op">$</span><span class="va">label.main</span>
<span class="op">)</span>

<span class="va">seurat</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">cell_type_singler_blueprintencode_main</span> <span class="op">&lt;-</span> <span class="va">singler_results_blueprintencode_main</span><span class="op">@</span><span class="va">listData</span><span class="op">$</span><span class="va">labels</span>

<span class="va">singler_scores</span> <span class="op">&lt;-</span> <span class="va">singler_results_blueprintencode_main</span><span class="op">@</span><span class="va">listData</span><span class="op">$</span><span class="va">scores</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>assigned_score <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>

<span class="kw">for</span> <span class="op">(</span> <span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">singler_scores</span><span class="op">)</span><span class="op">)</span> <span class="op">)</span> <span class="op">{</span>
  <span class="va">singler_scores</span><span class="op">$</span><span class="va">assigned_score</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">singler_scores</span><span class="op">[[</span><span class="va">singler_results_blueprintencode_main</span><span class="op">@</span><span class="va">listData</span><span class="op">$</span><span class="va">labels</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>
<span class="op">}</span>

<span class="va">seurat</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">cell_type_singler_blueprintencode_main_score</span> <span class="op">&lt;-</span> <span class="va">singler_scores</span><span class="op">$</span><span class="va">assigned_score</span></code></pre></div>
</div>
<div id="calculate-relationship-trees" class="section level1">
<h1 class="hasAnchor">
<a href="#calculate-relationship-trees" class="anchor"></a>Calculate relationship trees</h1>
<p>Here, we calculate phylogenetic trees that represent the similarity between different subgroups, in this case clusters and cell types. To do so, we need to set the identity in the Seurat object to the current grouping of interest, calculate the tree, and then copy the result to the <code>misc</code> slot in the Seurat object so it can be exported later.</p>
<div id="clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#clusters" class="anchor"></a>Clusters</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://satijalab.org/seurat/reference/Idents.html">Idents</a></span><span class="op">(</span><span class="va">seurat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"seurat_clusters"</span>
<span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/BuildClusterTree.html">BuildClusterTree</a></span><span class="op">(</span>
  <span class="va">seurat</span>,
  dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>,
  reorder <span class="op">=</span> <span class="cn">FALSE</span>,
  reorder.numeric <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span>
<span class="va">seurat</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">trees</span><span class="op">$</span><span class="va">seurat_clusters</span> <span class="op">&lt;-</span> <span class="va">seurat</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">BuildClusterTree</span></code></pre></div>
</div>
<div id="cell-types" class="section level2">
<h2 class="hasAnchor">
<a href="#cell-types" class="anchor"></a>Cell types</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://satijalab.org/seurat/reference/Idents.html">Idents</a></span><span class="op">(</span><span class="va">seurat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"cell_type_singler_blueprintencode_main"</span>
<span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/BuildClusterTree.html">BuildClusterTree</a></span><span class="op">(</span>
  <span class="va">seurat</span>,
  dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>,
  reorder <span class="op">=</span> <span class="cn">FALSE</span>,
  reorder.numeric <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span>
<span class="va">seurat</span><span class="op">@</span><span class="va">misc</span><span class="op">$</span><span class="va">trees</span><span class="op">$</span><span class="va">cell_type_singler_blueprintencode_main</span> <span class="op">&lt;-</span> <span class="va">seurat</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">BuildClusterTree</span></code></pre></div>
</div>
</div>
<div id="cerebroapp-functions" class="section level1">
<h1 class="hasAnchor">
<a href="#cerebroapp-functions" class="anchor"></a>cerebroApp functions</h1>
<p>Now we use a set of cerebroApp function to get more information about the data set and the contained clusters and cell types.</p>
<div id="calculate-percentage-of-mitochondrial-and-ribosomal-transcripts" class="section level2">
<h2 class="hasAnchor">
<a href="#calculate-percentage-of-mitochondrial-and-ribosomal-transcripts" class="anchor"></a>Calculate percentage of mitochondrial and ribosomal transcripts</h2>
<p>First, for every cell we calculate the percentage of mitochondrial and ribosomal transcripts of all transcripts using the <code><a href="../reference/addPercentMtRibo.html">addPercentMtRibo()</a></code> function.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addPercentMtRibo.html">addPercentMtRibo</a></span><span class="op">(</span>
  <span class="va">seurat</span>,
  organism <span class="op">=</span> <span class="st">'hg'</span>,
  gene_nomenclature <span class="op">=</span> <span class="st">'name'</span>
<span class="op">)</span>
<span class="co">#&gt; [17:55:31] No mitochondrial genes found in data set.</span>
<span class="co">#&gt; [17:55:31] Calculate percentage of 1 ribosomal transcript(s) present in the data set...</span></code></pre></div>
</div>
<div id="get-most-expressed-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#get-most-expressed-genes" class="anchor"></a>Get most expressed genes</h2>
<p>Another interesting perspective is to look at the 100 most expressed genes across all cells and for every cluster and cell type. This can help in understanding the transcriptional activity of a given cell population.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getMostExpressedGenes.html">getMostExpressedGenes</a></span><span class="op">(</span>
  <span class="va">seurat</span>,
  assay <span class="op">=</span> <span class="st">'RNA'</span>,
  groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'sample'</span>,<span class="st">'seurat_clusters'</span>,<span class="st">'cell_type_singler_blueprintencode_main'</span><span class="op">)</span>
<span class="op">)</span>
<span class="co">#&gt; [17:55:31] Group `sample` contains only one subgroup. Will calculate most expressed genes across all cells of this group...</span>
<span class="co">#&gt; [17:55:31] Get most expressed genes for 2 groups in `seurat_clusters`...</span>
<span class="co">#&gt; [17:55:31] Get most expressed genes for 8 groups in `cell_type_singler_blueprintencode_main`...</span></code></pre></div>
</div>
<div id="identify-marker-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-marker-genes" class="anchor"></a>Identify marker genes</h2>
<p>Marker genes are genes which are particularly strongly or weakly expressed in a given cell population, e.g. a cluster. This information can help to distinguish the role of different cell groups in the data set. We can identify marker genes using the <code><a href="../reference/getMarkerGenes.html">getMarkerGenes()</a></code> function which internally uses the <code><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html">Seurat::FindAllMarkers()</a></code> function but also stores the results in the Seurat object under the <code>misc</code> slot.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getMarkerGenes.html">getMarkerGenes</a></span><span class="op">(</span>
  <span class="va">seurat</span>,
  assay <span class="op">=</span> <span class="st">'RNA'</span>,
  organism <span class="op">=</span> <span class="st">'hg'</span>,
  groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'seurat_clusters'</span>,<span class="st">'cell_type_singler_blueprintencode_main'</span><span class="op">)</span>,
  name <span class="op">=</span> <span class="st">'cerebro_seurat'</span>,
  only_pos <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">#&gt; Ensembl site unresponsive, trying uswest mirror</span>
<span class="co">#&gt; [17:56:26] Get marker genes for 2 groups in `seurat_clusters`...</span>
<span class="co">#&gt; Calculating cluster 0</span>
<span class="co">#&gt; Calculating cluster 1</span>
<span class="co">#&gt; [17:56:26] Get marker genes for 8 groups in `cell_type_singler_blueprintencode_main`...</span>
<span class="co">#&gt; Calculating cluster CD4+ T-cells</span>
<span class="co">#&gt; Calculating cluster CD8+ T-cells</span>
<span class="co">#&gt; Calculating cluster B-cells</span>
<span class="co">#&gt; Calculating cluster Monocytes</span>
<span class="co">#&gt; Calculating cluster NK cells</span>
<span class="co">#&gt; Calculating cluster Erythrocytes</span>
<span class="co">#&gt; Calculating cluster Endothelial cells</span>
<span class="co">#&gt; Calculating cluster Fibroblasts</span></code></pre></div>
</div>
<div id="perform-pathway-enrichment-on-marker-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#perform-pathway-enrichment-on-marker-genes" class="anchor"></a>Perform pathway enrichment on marker genes</h2>
<p>Using the previously identified marker genes, we can perform a pathway enrichmeny analysis using the <code><a href="../reference/getEnrichedPathways.html">getEnrichedPathways()</a></code> function. The function will internally use the <a href="https://amp.pharm.mssm.edu/Enrichr/">Enrichr</a> API and store the results in the Seurat object under the <code>misc</code> slot. All we have to do is provide the name that we specified in <code><a href="../reference/getMarkerGenes.html">getMarkerGenes()</a></code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getEnrichedPathways.html">getEnrichedPathways</a></span><span class="op">(</span>
  <span class="va">seurat</span>,
  marker_genes_input <span class="op">=</span> <span class="st">'cerebro_seurat'</span>,
  adj_p_cutoff <span class="op">=</span> <span class="fl">0.01</span>,
  max_terms <span class="op">=</span> <span class="fl">100</span>
<span class="op">)</span>
<span class="co">#&gt; [17:56:26] Found 2 groups: seurat_clusters, cell_type_singler_blueprintencode_main</span>
<span class="co">#&gt; [17:56:26] Get enriched pathways for group `seurat_clusters`...</span>
<span class="co">#&gt; [17:56:30] 47 pathways passed the thresholds across all group levels and databases.</span>
<span class="co">#&gt; [17:56:30] Get enriched pathways for group `cell_type_singler_blueprintencode_main`...</span>
<span class="co">#&gt; [17:56:46] 311 pathways passed the thresholds across all group levels and databases.</span></code></pre></div>
</div>
<div id="gene-set-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-set-enrichment" class="anchor"></a>Gene set enrichment</h2>
<p>On top of querying the databases of the <a href="https://amp.pharm.mssm.edu/Enrichr/">Enrichr</a> service, one can also perform gene set enrichment analysis on gene sets provided as a <code>.gmt</code> file, e.g. from the <a href="https://www.gsea-msigdb.org/gsea/msigdb/index.jsp">MSigDB</a>. The <code><a href="../reference/performGeneSetEnrichmentAnalysis.html">performGeneSetEnrichmentAnalysis()</a></code> function uses the <a href="https://bioconductor.org/packages/release/bioc/html/GSVA.html">GSVA</a> method in combination with additional statistics as published by <a href="https://f1000research.com/articles/8-296">Diaz-Mejia et. al.</a>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">example_gene_set</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/example_gene_set.gmt"</span>, package <span class="op">=</span> <span class="st">"cerebroApp"</span><span class="op">)</span>

<span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/performGeneSetEnrichmentAnalysis.html">performGeneSetEnrichmentAnalysis</a></span><span class="op">(</span>
  <span class="va">seurat</span>,
  assay <span class="op">=</span> <span class="st">'RNA'</span>,
  GMT_file <span class="op">=</span> <span class="va">example_gene_set</span>,
  groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'seurat_clusters'</span>,<span class="st">'cell_type_singler_blueprintencode_main'</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
</div>
</div>
<div id="trajectory-analysis-with-monocle-optional" class="section level1">
<h1 class="hasAnchor">
<a href="#trajectory-analysis-with-monocle-optional" class="anchor"></a>Trajectory analysis with Monocle (optional)</h1>
<p>Then, we perform trajectory analysis with <a href="https://cole-trapnell-lab.github.io/monocle-release/">Monocle v2</a> using the previously identified highly variable genes. We extract the trajectory from the generated Monocle object with the <code><a href="../reference/extractMonocleTrajectory.html">extractMonocleTrajectory()</a></code> function of cerebroApp and attach it to our Seurat object.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">monocle</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/monocle/man/newCellDataSet.html">newCellDataSet</a></span><span class="op">(</span>
  <span class="va">seurat</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span>,
  phenoData <span class="op">=</span> <span class="fu">new</span><span class="op">(</span><span class="st">'AnnotatedDataFrame'</span>, data <span class="op">=</span> <span class="va">seurat</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span>,
  featureData <span class="op">=</span> <span class="fu">new</span><span class="op">(</span><span class="st">'AnnotatedDataFrame'</span>, data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
    gene_short_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">seurat</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span><span class="op">)</span>,
    row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">seurat</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span>
  <span class="op">)</span>
<span class="op">)</span>

<span class="va">monocle</span> <span class="op">&lt;-</span> <span class="fu">estimateSizeFactors</span><span class="op">(</span><span class="va">monocle</span><span class="op">)</span>
<span class="va">monocle</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/monocle/man/setOrderingFilter.html">setOrderingFilter</a></span><span class="op">(</span><span class="va">monocle</span>, <span class="va">seurat</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">var.features</span><span class="op">)</span>
<span class="va">monocle</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/monocle/man/reduceDimension.html">reduceDimension</a></span><span class="op">(</span><span class="va">monocle</span>, max_components <span class="op">=</span> <span class="fl">2</span>, method <span class="op">=</span> <span class="st">'DDRTree'</span><span class="op">)</span>
<span class="va">monocle</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/monocle/man/orderCells.html">orderCells</a></span><span class="op">(</span><span class="va">monocle</span><span class="op">)</span>
<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="co">#&gt; Warning in if (class(projection) != "matrix") projection &lt;- as.matrix(projection): the condition has</span>
<span class="co">#&gt; length &gt; 1 and only the first element will be used</span>

<span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extractMonocleTrajectory.html">extractMonocleTrajectory</a></span><span class="op">(</span><span class="va">monocle</span>, <span class="va">seurat</span>, <span class="st">'highly_variable_genes'</span><span class="op">)</span></code></pre></div>
</div>
<div id="export-to-cerebro-format" class="section level1">
<h1 class="hasAnchor">
<a href="#export-to-cerebro-format" class="anchor"></a>Export to Cerebro format</h1>
<p>Finally, we use the <code><a href="../reference/exportFromSeurat.html">exportFromSeurat()</a></code> function of cerebroApp to export our Seurat object to a <code>.crb</code> file which can be loaded into Cerebro.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/exportFromSeurat.html">exportFromSeurat</a></span><span class="op">(</span>
  <span class="va">seurat</span>,
  assay <span class="op">=</span> <span class="st">'RNA'</span>,
  slot <span class="op">=</span> <span class="st">'data'</span>,
  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">'cerebro_pbmc_seurat_'</span>, <span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html">Sys.Date</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">'.crb'</span><span class="op">)</span>,
  experiment_name <span class="op">=</span> <span class="st">'pbmc'</span>,
  organism <span class="op">=</span> <span class="st">'hg'</span>,
  groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'sample'</span>,<span class="st">'seurat_clusters'</span>,<span class="st">'cell_type_singler_blueprintencode_main'</span><span class="op">)</span>,
  nUMI <span class="op">=</span> <span class="st">'nCount_RNA'</span>,
  nGene <span class="op">=</span> <span class="st">'nFeature_RNA'</span>,
  add_all_meta_data <span class="op">=</span> <span class="cn">TRUE</span>,
  verbose <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span>
<span class="co">#&gt; [17:56:48] Start collecting data...</span>
<span class="co">#&gt; [17:56:48] Overview of Cerebro object:</span>
<span class="co">#&gt; class: Cerebro_v1.3</span>
<span class="co">#&gt; cerebroApp version: 1.3.1</span>
<span class="co">#&gt; experiment name: pbmc</span>
<span class="co">#&gt; organism: hg</span>
<span class="co">#&gt; date of analysis: 2021-03-12</span>
<span class="co">#&gt; date of export: 2021-03-12</span>
<span class="co">#&gt; number of cells: 78</span>
<span class="co">#&gt; number of genes: 220</span>
<span class="co">#&gt; grouping variables (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main</span>
<span class="co">#&gt; cell cycle variables (0): </span>
<span class="co">#&gt; projections (2): UMAP, UMAP_3D</span>
<span class="co">#&gt; trees (2): seurat_clusters, cell_type_singler_blueprintencode_main</span>
<span class="co">#&gt; most expressed genes: sample, seurat_clusters, cell_type_singler_blueprintencode_main</span>
<span class="co">#&gt; marker genes:</span>
<span class="co">#&gt;   - cerebro_seurat (2): seurat_clusters, cell_type_singler_blueprintencode_main</span>
<span class="co">#&gt; enriched pathways:</span>
<span class="co">#&gt;   - cerebro_seurat_enrichr (2): seurat_clusters, cell_type_singler_blueprintencode_main</span>
<span class="co">#&gt; trajectories:</span>
<span class="co">#&gt;   - monocle2 (1): highly_variable_genes</span>
<span class="co">#&gt; extra material:</span>
<span class="co">#&gt; [17:56:48] Saving Cerebro object to: cerebro_pbmc_seurat_2021-03-12.crb</span>
<span class="co">#&gt; [17:56:48] Done!</span></code></pre></div>
<p>The Cerebro use interface can be launched using the <code><a href="../reference/launchCerebro.html">launchCerebro()</a></code> function.</p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Big Sur 10.16</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt;  [1] splines   parallel  stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [10] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] celldex_1.0.0               cerebroApp_1.3.1            monocle_2.18.0             </span>
<span class="co">#&gt;  [4] DDRTree_0.1.5               irlba_2.3.3                 VGAM_1.1-5                 </span>
<span class="co">#&gt;  [7] ggplot2_3.3.2               Matrix_1.3-0                SingleR_1.4.1              </span>
<span class="co">#&gt; [10] SummarizedExperiment_1.20.0 Biobase_2.50.0              GenomicRanges_1.42.0       </span>
<span class="co">#&gt; [13] GenomeInfoDb_1.26.2         IRanges_2.24.1              S4Vectors_0.28.1           </span>
<span class="co">#&gt; [16] BiocGenerics_0.36.0         MatrixGenerics_1.2.0        matrixStats_0.57.0         </span>
<span class="co">#&gt; [19] Seurat_3.2.3                dplyr_1.0.2                 BiocManager_1.30.10        </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] rappdirs_0.3.1                scattermore_0.7               ragg_1.1.0                   </span>
<span class="co">#&gt;   [4] tidyr_1.1.2                   bit64_4.0.5                   knitr_1.30                   </span>
<span class="co">#&gt;   [7] DelayedArray_0.16.1           data.table_1.13.4             rpart_4.1-15                 </span>
<span class="co">#&gt;  [10] RCurl_1.98-1.2                generics_0.1.0                callr_3.5.1                  </span>
<span class="co">#&gt;  [13] cowplot_1.1.1                 usethis_2.0.0                 RSQLite_2.2.1                </span>
<span class="co">#&gt;  [16] RANN_2.6.1                    combinat_0.0-8                proxy_0.4-25                 </span>
<span class="co">#&gt;  [19] future_1.21.0                 bit_4.0.4                     spatstat.data_2.0-0          </span>
<span class="co">#&gt;  [22] xml2_1.3.2                    httpuv_1.5.4                  assertthat_0.2.1             </span>
<span class="co">#&gt;  [25] viridis_0.5.1                 xfun_0.19                     hms_0.5.3                    </span>
<span class="co">#&gt;  [28] evaluate_0.14                 promises_1.1.1                fansi_0.4.1                  </span>
<span class="co">#&gt;  [31] progress_1.2.2                dbplyr_2.0.0                  igraph_1.2.6                 </span>
<span class="co">#&gt;  [34] DBI_1.1.0                     htmlwidgets_1.5.3             sparsesvd_0.2                </span>
<span class="co">#&gt;  [37] purrr_0.3.4                   ellipsis_0.3.1                RSpectra_0.16-0              </span>
<span class="co">#&gt;  [40] annotate_1.68.0               biomaRt_2.46.0                deldir_0.2-10                </span>
<span class="co">#&gt;  [43] sparseMatrixStats_1.2.1       vctrs_0.3.6                   remotes_2.2.0                </span>
<span class="co">#&gt;  [46] ROCR_1.0-11                   abind_1.4-5                   withr_2.3.0                  </span>
<span class="co">#&gt;  [49] sctransform_0.3.2             prettyunits_1.1.1             goftest_1.2-2                </span>
<span class="co">#&gt;  [52] cluster_2.1.0                 ExperimentHub_1.16.0          ape_5.4-1                    </span>
<span class="co">#&gt;  [55] lazyeval_0.2.2                crayon_1.3.4                  pkgconfig_2.0.3              </span>
<span class="co">#&gt;  [58] slam_0.1-48                   nlme_3.1-151                  pkgload_1.1.0                </span>
<span class="co">#&gt;  [61] devtools_2.3.2                rlang_0.4.9                   globals_0.14.0               </span>
<span class="co">#&gt;  [64] lifecycle_0.2.0               miniUI_0.1.1.1                colourpicker_1.1.0           </span>
<span class="co">#&gt;  [67] BiocFileCache_1.14.0          rsvd_1.0.3                    AnnotationHub_2.22.0         </span>
<span class="co">#&gt;  [70] rprojroot_2.0.2               polyclip_1.10-0               GSVA_1.38.0                  </span>
<span class="co">#&gt;  [73] lmtest_0.9-38                 shinyFiles_0.9.0              graph_1.68.0                 </span>
<span class="co">#&gt;  [76] zoo_1.8-8                     ggridges_0.5.3                processx_3.4.5               </span>
<span class="co">#&gt;  [79] pheatmap_1.0.12               png_0.1-7                     viridisLite_0.3.0            </span>
<span class="co">#&gt;  [82] bitops_1.0-6                  shinydashboard_0.7.1          KernSmooth_2.23-18           </span>
<span class="co">#&gt;  [85] blob_1.2.1                    DelayedMatrixStats_1.12.3     stringr_1.4.0                </span>
<span class="co">#&gt;  [88] qvalue_2.22.0                 parallelly_1.22.0             readr_1.4.0                  </span>
<span class="co">#&gt;  [91] beachmat_2.6.4                scales_1.1.1                  memoise_1.1.0                </span>
<span class="co">#&gt;  [94] GSEABase_1.52.1               magrittr_2.0.1                plyr_1.8.6                   </span>
<span class="co">#&gt;  [97] ica_1.0-2                     zlibbioc_1.36.0               compiler_4.0.3               </span>
<span class="co">#&gt; [100] HSMMSingleCell_1.10.0         RColorBrewer_1.1-2            fitdistrplus_1.1-3           </span>
<span class="co">#&gt; [103] cli_2.2.0                     XVector_0.30.0                listenv_0.8.0                </span>
<span class="co">#&gt; [106] patchwork_1.1.1               pbapply_1.4-3                 ps_1.5.0                     </span>
<span class="co">#&gt; [109] MASS_7.3-53                   mgcv_1.8-33                   tidyselect_1.1.0             </span>
<span class="co">#&gt; [112] stringi_1.5.3                 textshaping_0.3.0             densityClust_0.3             </span>
<span class="co">#&gt; [115] yaml_2.2.1                    BiocSingular_1.6.0            askpass_1.1                  </span>
<span class="co">#&gt; [118] ggrepel_0.9.1                 grid_4.0.3                    tools_4.0.3                  </span>
<span class="co">#&gt; [121] future.apply_1.6.0            gridExtra_2.3                 Rtsne_0.15                   </span>
<span class="co">#&gt; [124] digest_0.6.27                 FNN_1.1.3                     shiny_1.5.0                  </span>
<span class="co">#&gt; [127] qlcMatrix_0.9.7               Rcpp_1.0.5                    BiocVersion_3.12.0           </span>
<span class="co">#&gt; [130] later_1.1.0.1                 RcppAnnoy_0.0.18              shinyWidgets_0.5.4           </span>
<span class="co">#&gt; [133] httr_1.4.2                    AnnotationDbi_1.52.0          colorspace_2.0-0             </span>
<span class="co">#&gt; [136] XML_3.99-0.5                  fs_1.5.0                      tensor_1.5                   </span>
<span class="co">#&gt; [139] reticulate_1.18               uwot_0.1.10                   spatstat.utils_2.0-0         </span>
<span class="co">#&gt; [142] pkgdown_1.6.1                 plotly_4.9.2.2                sessioninfo_1.1.1            </span>
<span class="co">#&gt; [145] systemfonts_1.0.1             xtable_1.8-4                  jsonlite_1.7.2               </span>
<span class="co">#&gt; [148] spatstat_1.64-1               testthat_3.0.1                R6_2.5.0                     </span>
<span class="co">#&gt; [151] pillar_1.4.7                  htmltools_0.5.1.1             mime_0.9                     </span>
<span class="co">#&gt; [154] glue_1.4.2                    fastmap_1.0.1                 DT_0.16                      </span>
<span class="co">#&gt; [157] BiocParallel_1.24.1           BiocNeighbors_1.8.2           interactiveDisplayBase_1.28.0</span>
<span class="co">#&gt; [160] codetools_0.2-18              pkgbuild_1.2.0                lattice_0.20-41              </span>
<span class="co">#&gt; [163] tibble_3.0.4                  curl_4.3                      leiden_0.3.7                 </span>
<span class="co">#&gt; [166] shinyjs_2.0.0                 openssl_1.4.3                 survival_3.2-7               </span>
<span class="co">#&gt; [169] limma_3.46.0                  rmarkdown_2.6                 docopt_0.7.1                 </span>
<span class="co">#&gt; [172] desc_1.2.0                    fastICA_1.2-2                 munsell_0.5.0                </span>
<span class="co">#&gt; [175] GenomeInfoDbData_1.2.4        reshape2_1.4.4                gtable_0.3.0                 </span>
<span class="co">#&gt; [178] shinycssloaders_1.0.0         msigdbr_7.2.1</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Roman Hillje.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
