<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Export and visualize custom tables and plots • cerebroApp</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Export and visualize custom tables and plots">
<meta property="og:description" content="cerebroApp">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cerebroApp</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cerebroApp_workflow_Seurat.html">The cerebroApp workflow (Seurat)</a>
    </li>
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">The Cerebro interface</a>
      <ul class="dropdown-menu" role="menu">
<li>
          <a href="../articles/Cerebro_interface_v1.2.html">Cerebro v1.2 and older</a>
        </li>
      </ul>
</li>
    <li>
      <a href="../articles/export_a_data_set_in_SCE_format.html">Export an SCE/SingleCellExperiment object</a>
    </li>
    <li>
      <a href="../articles/export_and_visualize_custom_tables_and_plots.html">Export and visualize custom tables and plots</a>
    </li>
    <li>
      <a href="../articles/launch_cerebro_with_pre-loaded_data_set.html">Launch Cerebro with pre-loaded data set</a>
    </li>
    <li>
      <a href="../articles/host_cerebro_on_shinyapps.html">Host Cerebro on shinyapps.io</a>
    </li>
    <li>
      <a href="../articles/control_access_to_cerebro_with_a_login_page.html">Control access to Cerebro with a login page</a>
    </li>
    <li>
      <a href="../articles/overview_of_cerebro_v1.3_class.html">Overview of the Cerebro v1.3 class</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Release notes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/release_notes_v1.3.html">v1.3</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/romanhaa/cerebroApp/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/fakechek1">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="export_and_visualize_custom_tables_and_plots_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Export and visualize custom tables and plots</h1>
                        <h4 class="author">Roman Hillje</h4>
            
            <h4 class="date">25 September, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/romanhaa/cerebroApp/blob/master/vignettes/export_and_visualize_custom_tables_and_plots.Rmd"><code>vignettes/export_and_visualize_custom_tables_and_plots.Rmd</code></a></small>
      <div class="hidden name"><code>export_and_visualize_custom_tables_and_plots.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>cerebroApp v1.3 provides new possibilities to export and visualize custom data that you would like to share alongside with your scRNA-seq data. At the time of writing, you can attach tables and plots (made with ggplot2) to your data set. In this article, I will show you to do it.</p>
</div>
<div id="tables" class="section level1">
<h1 class="hasAnchor">
<a href="#tables" class="anchor"></a>Tables</h1>
<p>In previous cerebroApp versions, a very specific format for tables containing marker genes and enriched pathways was expected, resulting in a severe limitation of which methods you can use to generate the results, or alternatively requiring manual modification of the table to fit the format. Since cerebroApp v1.3, it is possible to export and visualize tables of any format in Cerebro, as long as they are a <code>data.frame</code>. Due to dynamic column formatting and color highlighting, the table content can be visually enhanced anyway.</p>
<div id="marker-genes-and-enriched-pathways" class="section level2">
<h2 class="hasAnchor">
<a href="#marker-genes-and-enriched-pathways" class="anchor"></a>Marker genes and enriched pathways</h2>
<p>Let’s assume you have a <code>Seurat</code> object but generated tables of differentially expressed genes and enriched pathways using other tools/methods than those built into cerebroApp. To export those tables, you just need to put it in the right place, following a “method” and “name” scheme. In this example, we assume you have a table of marker genes called <code>custom_table</code> that looks like this:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">tibble</span>)

<span class="no">custom_table</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tribble.html">tribble</a></span>(
      ~<span class="no">a</span>,   ~<span class="no">b</span>,  ~<span class="no">c</span>,       ~<span class="no">d</span>,      ~<span class="no">e</span>,
  <span class="st">"this"</span>, <span class="st">"is"</span>, <span class="st">"a"</span>, <span class="st">"custom"</span>, <span class="st">"table"</span>
)</pre></body></html></div>
<p>To make sure it is exported when running <code><a href="../reference/exportFromSeurat.html">exportFromSeurat()</a></code>, simply save it in a list with a name for both the method and the table, e.g. <code>custom_method</code> and <code>custom_table</code>:</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="no">pbmc_Seurat</span>@<span class="kw">misc</span>$<span class="no">marker_genes</span><span class="kw">[[</span><span class="st">'custom_method'</span>]] <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(
  <span class="st">"custom_table"</span> <span class="kw">=</span> <span class="no">custom_table</span>
)</pre></body></html></div>
<p>Then, you can export your data set:</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="../reference/exportFromSeurat.html">exportFromSeurat</a></span>(
  <span class="no">pbmc_Seurat</span>,
  <span class="kw">assay</span> <span class="kw">=</span> <span class="st">'SCT'</span>,
  <span class="kw">slot</span> <span class="kw">=</span> <span class="st">'data'</span>,
  <span class="kw">file</span> <span class="kw">=</span> <span class="st">'~/Dropbox/Cerebro_development/pbmc_Seurat.crb'</span>,
  <span class="kw">experiment_name</span> <span class="kw">=</span> <span class="st">'pbmc_Seurat'</span>,
  <span class="kw">organism</span> <span class="kw">=</span> <span class="st">'hg'</span>,
  <span class="kw">groups</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'sample'</span>,<span class="st">'seurat_clusters'</span>,<span class="st">'cell_type_singler_blueprintencode_main'</span>),
  <span class="kw">cell_cycle</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'cell_cycle_seurat'</span>),
  <span class="kw">nUMI</span> <span class="kw">=</span> <span class="st">'nCount_RNA'</span>,
  <span class="kw">nGene</span> <span class="kw">=</span> <span class="st">'nFeature_RNA'</span>,
  <span class="kw">add_all_meta_data</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)</pre></body></html></div>
<pre><code>[15:28:21] Start collecting data...
[15:28:21] Overview of Cerebro object:

class: Cerebro_v1.3
cerebroApp version: 1.3.0
experiment name: pbmc_Seurat
organism: hg
date of analysis: 2020-02-19
date of export: 2020-09-10
number of cells: 5,697
number of genes: 15,907
grouping variables (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
cell cycle variables (1): cell_cycle_seurat
projections (2): UMAP, UMAP_3D
trees (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
most expressed genes: sample, seurat_clusters, cell_type_singler_blueprintencode_main
marker genes:
 - cerebro_seurat (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main, 
 - custom_method (1): custom_table
enriched pathways:
 - cerebro_seurat_enrichr (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main, 
 - cerebro_GSVA (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
trajectories:
 - monocle2 (1): highly_variable_genes

[15:28:21] Saving Cerebro object to: ~/Dropbox/Cerebro_development/pbmc_Seurat.crb
[15:28:27] Done!</code></pre>
<p>You can see that among the marker gene results is the <code>custom_method</code> and <code>custom_table</code>.</p>
<p>Then, after loading the data into Cerebro, we will see this:</p>
<p><a target="_blank" href="export_and_visualize_custom_tables_and_plots_files/table_marker_genes.png"><img src="export_and_visualize_custom_tables_and_plots_files/table_marker_genes.png" alt="Custom marker gene table in Cerebro" width="100%" height="auto"></a></p>
<p>Granted, this isn’t a very pretty table, but you get the idea.</p>
</div>
<div id="extra-material" class="section level2">
<h2 class="hasAnchor">
<a href="#extra-material" class="anchor"></a>Extra material</h2>
<p>When you have tables that might be useful to interpret the data set but don’t contain marker genes, differentially expressed genes, or enriched pathways, then you can add them as extra material. Again, just make sure you follow the general structure. In the case of a <code>Seurat</code> object, store the tables with an interpretable name in a list at <code>@misc$extra_material$tables</code>. You can see an example below, where we store the same example table as before:</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">custom_table</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span>(
  <span class="kw">a</span> <span class="kw">=</span> <span class="st">"this"</span>,
  <span class="kw">b</span> <span class="kw">=</span> <span class="st">"is"</span>,
  <span class="kw">c</span> <span class="kw">=</span> <span class="st">"a"</span>,
  <span class="kw">d</span> <span class="kw">=</span> <span class="st">"test"</span>
)

<span class="no">pbmc_Seurat</span>@<span class="kw">misc</span>$<span class="no">extra_material</span>$<span class="no">tables</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(
  <span class="st">"test"</span> <span class="kw">=</span> <span class="no">custom_table</span>
)

<span class="fu"><a href="../reference/exportFromSeurat.html">exportFromSeurat</a></span>(
  <span class="no">pbmc_Seurat</span>,
  <span class="kw">assay</span> <span class="kw">=</span> <span class="st">'SCT'</span>,
  <span class="kw">slot</span> <span class="kw">=</span> <span class="st">'data'</span>,
  <span class="kw">file</span> <span class="kw">=</span> <span class="st">'~/Dropbox/Cerebro_development/pbmc_Seurat.crb'</span>,
  <span class="kw">experiment_name</span> <span class="kw">=</span> <span class="st">'pbmc_Seurat'</span>,
  <span class="kw">organism</span> <span class="kw">=</span> <span class="st">'hg'</span>,
  <span class="kw">groups</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'sample'</span>,<span class="st">'seurat_clusters'</span>,<span class="st">'cell_type_singler_blueprintencode_main'</span>),
  <span class="kw">cell_cycle</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'cell_cycle_seurat'</span>),
  <span class="kw">nUMI</span> <span class="kw">=</span> <span class="st">'nCount_RNA'</span>,
  <span class="kw">nGene</span> <span class="kw">=</span> <span class="st">'nFeature_RNA'</span>,
  <span class="kw">add_all_meta_data</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)</pre></body></html></div>
<pre><code>[15:07:48] Start collecting data...
[15:07:50] Overview of Cerebro object:

class: Cerebro_v1.3
cerebroApp version: 1.3.0
experiment name: pbmc_Seurat
organism: hg
date of analysis: 2020-02-19
date of export: 2020-09-11
number of cells: 5,697
number of genes: 15,907
grouping variables (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
cell cycle variables (1): cell_cycle_seurat
projections (2): UMAP, UMAP_3D
trees (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
most expressed genes: sample, seurat_clusters, cell_type_singler_blueprintencode_main
marker genes:
  - cerebro_seurat (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main, 
  - custom_method (1): custom_table
enriched pathways:
  - cerebro_seurat_enrichr (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main, 
  - cerebro_GSVA (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
trajectories:
  - monocle2 (1): highly_variable_genes
extra material:
  - tables (1): test

[15:07:50] Saving Cerebro object to: ~/Dropbox/Cerebro_development/pbmc_Seurat.crb
[15:07:55] Done!</code></pre>
<p>This time, you see that our table with the name “test” was exported as part of the extra material. In Cerebro, when a data set contains extra material, a tab will appear in the side bar which will give you access to the table, as shown in the screenshot:</p>
<p><a target="_blank" href="export_and_visualize_custom_tables_and_plots_files/table_extra_material.png"><img src="export_and_visualize_custom_tables_and_plots_files/table_extra_material.png" alt="Custom table as extra material in Cerebro" width="100%" height="auto"></a></p>
<p>Again, this table isn’t particularly informative, but hopefully you got an idea of the procedure.</p>
</div>
</div>
<div id="plots" class="section level1">
<h1 class="hasAnchor">
<a href="#plots" class="anchor"></a>Plots</h1>
<p>Custom plots can be exported and visualized in the “Extra material” tab as well. The only requirement is that they were produced with ggplot2. You can switch between an interactive version of the plot (converted with plotly) or the unmodified plot.</p>
<div id="example-1" class="section level2">
<h2 class="hasAnchor">
<a href="#example-1" class="anchor"></a>Example #1</h2>
<p>Let’s look at an example in which we create a simple plot with ggplot2, call it <code>iris</code>, and add it to the Seurat object:</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)

<span class="no">custom_plot</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">iris</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="no">Sepal.Length</span>, <span class="no">Sepal.Width</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="no">Species</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>()

<span class="no">pbmc_Seurat</span>@<span class="kw">misc</span>$<span class="no">extra_material</span>$<span class="no">plots</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(
  <span class="st">"iris"</span> <span class="kw">=</span> <span class="no">custom_plot</span>
)

<span class="fu"><a href="../reference/exportFromSeurat.html">exportFromSeurat</a></span>(
  <span class="no">pbmc_Seurat</span>,
  <span class="kw">assay</span> <span class="kw">=</span> <span class="st">'SCT'</span>,
  <span class="kw">slot</span> <span class="kw">=</span> <span class="st">'data'</span>,
  <span class="kw">file</span> <span class="kw">=</span> <span class="st">'~/Dropbox/Cerebro_development/pbmc_Seurat.crb'</span>,
  <span class="kw">experiment_name</span> <span class="kw">=</span> <span class="st">'pbmc_Seurat'</span>,
  <span class="kw">organism</span> <span class="kw">=</span> <span class="st">'hg'</span>,
  <span class="kw">groups</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'sample'</span>,<span class="st">'seurat_clusters'</span>,<span class="st">'cell_type_singler_blueprintencode_main'</span>),
  <span class="kw">cell_cycle</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'cell_cycle_seurat'</span>),
  <span class="kw">nUMI</span> <span class="kw">=</span> <span class="st">'nCount_RNA'</span>,
  <span class="kw">nGene</span> <span class="kw">=</span> <span class="st">'nFeature_RNA'</span>,
  <span class="kw">add_all_meta_data</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)</pre></body></html></div>
<pre><code>[11:54:33] Start collecting data...
[11:54:33] Overview of Cerebro object:

class: Cerebro_v1.3
cerebroApp version: 1.3.0
experiment name: pbmc_Seurat
organism: hg
date of analysis: 2020-02-19
date of export: 2020-09-24
number of cells: 5,697
number of genes: 15,907
grouping variables (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
cell cycle variables (1): cell_cycle_seurat
projections (2): UMAP, UMAP_3D
trees (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
most expressed genes: sample, seurat_clusters, cell_type_singler_blueprintencode_main
marker genes:
  - cerebro_seurat (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
enriched pathways:
  - cerebro_seurat_enrichr (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main, 
  - cerebro_GSVA (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
trajectories:
  - monocle2 (1): highly_variable_genes
extra material:
  - plots (1): iris

[11:54:33] Saving Cerebro object to: ~/Dropbox/Cerebro_development/pbmc_Seurat_dgCMatrix.crb
[11:54:37] Done!</code></pre>
<p>In the log messages of the export process, we see that the plot was correctly extracted and is listed in the “extra material”.</p>
<p>In Cerebro, we can now access this table in the “Extra material” tab through the “plots” category.</p>
<p><a target="_blank" href="export_and_visualize_custom_tables_and_plots_files/plot_extra_material_1.png"><img src="export_and_visualize_custom_tables_and_plots_files/plot_extra_material_1.png" alt="Custom plot as extra material in Cerebro" width="100%" height="auto"></a></p>
</div>
<div id="example-2" class="section level2">
<h2 class="hasAnchor">
<a href="#example-2" class="anchor"></a>Example #2</h2>
<p>In the second example, we prepare a dot plot showing the expression of the most significant marker gene for each cluster.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">dplyr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">tidyr</span>)

<span class="co"># cells will be grouped by clusters that they have been assigned to</span>
<span class="no">cluster_ids</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span>(<span class="no">seurat</span>@<span class="kw">meta.data</span>$<span class="no">seurat_clusters</span>)

<span class="co"># select a set of genes for which we want to show expression</span>
<span class="no">genes_to_show</span> <span class="kw">&lt;-</span> <span class="no">seurat</span>@<span class="kw">misc</span>$<span class="no">marker_genes</span>$<span class="no">cerebro_seurat</span>$<span class="no">seurat_clusters</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span>(<span class="no">seurat_clusters</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(<span class="no">p_val_adj</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice</a></span>(<span class="fl">1</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>(<span class="no">gene</span>)

<span class="co"># for every cluster-gene combination, calculate the average expression across</span>
<span class="co"># all cells and then transform the data into a data frame</span>
<span class="no">expression_levels_per_cluster</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">vapply</a></span>(
    <span class="no">cluster_ids</span>, <span class="kw">FUN.VALUE</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">numeric</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">cluster_ids</span>)), <span class="kw">function</span>(<span class="no">x</span>) {
      <span class="no">cells_in_current_cluster</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">seurat</span>@<span class="kw">meta.data</span>$<span class="no">seurat_cluster</span> <span class="kw">==</span> <span class="no">x</span>)
      <span class="kw pkg">Matrix</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html">rowMeans</a></span>(<span class="no">seurat</span>@<span class="kw">assays</span>$<span class="no">SCT</span>@<span class="kw">data</span>[<span class="no">genes_to_show</span>,<span class="no">cells_in_current_cluster</span>])
    }
  ) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>() <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>() <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="kw">cluster</span> <span class="kw">=</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames</a></span>(<span class="no">.</span>)) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">cluster</span>, <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">everything</a></span>()) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span>(
    <span class="kw">cols</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">2</span>:<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">.</span>)),
    <span class="kw">names_to</span> <span class="kw">=</span> <span class="st">'gene'</span>
  ) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span>(<span class="kw">expression</span> <span class="kw">=</span> <span class="no">value</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="kw">id_to_merge</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="no">cluster</span>, <span class="st">'_'</span>, <span class="no">gene</span>))

<span class="co"># for every cluster-gene combination, calculate the percentage of cells in the</span>
<span class="co"># respective group that has at least 1 transcript (this means we consider it</span>
<span class="co"># as expressing the gene) and then transform the data into a data frame</span>
<span class="no">percentage_of_cells_expressing_gene</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">vapply</a></span>(
    <span class="no">cluster_ids</span>, <span class="kw">FUN.VALUE</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">numeric</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">cluster_ids</span>)), <span class="kw">function</span>(<span class="no">x</span>) {
      <span class="no">cells_in_current_cluster</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">seurat</span>@<span class="kw">meta.data</span>$<span class="no">seurat_cluster</span> <span class="kw">==</span> <span class="no">x</span>)
      <span class="kw pkg">Matrix</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html">rowSums</a></span>(<span class="no">seurat</span>@<span class="kw">assays</span>$<span class="no">SCT</span>@<span class="kw">data</span>[<span class="no">genes_to_show</span>,<span class="no">cells_in_current_cluster</span>] <span class="kw">!=</span> <span class="fl">0</span>)
    }
  ) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>() <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>() <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="kw">cluster</span> <span class="kw">=</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames</a></span>(<span class="no">.</span>)) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">cluster</span>, <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">everything</a></span>()) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span>(
    <span class="kw">cols</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">2</span>:<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">.</span>)),
    <span class="kw">names_to</span> <span class="kw">=</span> <span class="st">'gene'</span>
  ) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span>(<span class="kw">cell_count</span> <span class="kw">=</span> <span class="no">value</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span>(
    <span class="no">.</span>,
    <span class="no">seurat</span>@<span class="kw">meta.data</span> <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span>(<span class="no">seurat_clusters</span>) <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">tally</a></span>() <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span>(<span class="kw">cluster</span> <span class="kw">=</span> <span class="no">seurat_clusters</span>),
    <span class="kw">by</span> <span class="kw">=</span> <span class="st">'cluster'</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(
    <span class="kw">id_to_merge</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="no">cluster</span>, <span class="st">'_'</span>, <span class="no">gene</span>),
    <span class="kw">percent_cells</span> <span class="kw">=</span> <span class="no">cell_count</span> / <span class="no">n</span>
  )

<span class="co"># merge the two data frames created before and plot the data</span>
<span class="no">custom_plot</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span>(
    <span class="no">expression_levels_per_cluster</span>,
    <span class="no">percentage_of_cells_expressing_gene</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">id_to_merge</span>, <span class="no">percent_cells</span>),
    <span class="kw">by</span> <span class="kw">=</span> <span class="st">'id_to_merge'</span>
  ) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(
    <span class="kw">cluster</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="no">cluster</span>, <span class="kw">levels</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(<span class="no">cluster_ids</span>)),
    <span class="kw">gene</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="no">gene</span>, <span class="kw">levels</span> <span class="kw">=</span> <span class="no">genes_to_show</span>)
  ) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="no">gene</span>, <span class="no">cluster</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">color</span> <span class="kw">=</span> <span class="no">expression</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="no">percent_cells</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_color_distiller</a></span>(
    <span class="kw">palette</span> <span class="kw">=</span> <span class="st">'Reds'</span>,
    <span class="kw">direction</span> <span class="kw">=</span> <span class="fl">1</span>,
    <span class="kw">name</span> <span class="kw">=</span> <span class="st">'Log-normalised\nexpression'</span>,
    <span class="kw">guide</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_colourbar.html">guide_colorbar</a></span>(<span class="kw">frame.colour</span> <span class="kw">=</span> <span class="st">"black"</span>, <span class="kw">ticks.colour</span> <span class="kw">=</span> <span class="st">"black"</span>)
  ) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html">scale_size</a></span>(<span class="kw">name</span> <span class="kw">=</span> <span class="st">'Percent\nof cells'</span>, <span class="kw">labels</span> <span class="kw">=</span> <span class="kw pkg">scales</span><span class="kw ns">::</span><span class="no"><a href="https://scales.r-lib.org//reference/label_percent.html">percent</a></span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="kw">y</span> <span class="kw">=</span> <span class="st">'Cluster'</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">'Expression'</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html">coord_fixed</a></span>() +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>() +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(
    <span class="kw">axis.title.x</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span>(),
    <span class="kw">axis.text.x</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="kw">angle</span> <span class="kw">=</span> <span class="fl">45</span>, <span class="kw">hjust</span> <span class="kw">=</span> <span class="fl">1</span>)
  )</pre></body></html></div>
<p>With the plot ready, we can store it in the Seurat object and export it as before.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">seurat</span>@<span class="kw">misc</span>$<span class="no">extra_material</span>$<span class="no">plots</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(
  <span class="st">"dot_plot_marker_genes_clusters"</span> <span class="kw">=</span> <span class="no">custom_plot</span>
)

<span class="fu"><a href="../reference/exportFromSeurat.html">exportFromSeurat</a></span>(
  <span class="no">seurat</span>,
  <span class="kw">assay</span> <span class="kw">=</span> <span class="st">'SCT'</span>,
  <span class="kw">slot</span> <span class="kw">=</span> <span class="st">'data'</span>,
  <span class="kw">file</span> <span class="kw">=</span> <span class="st">'~/Dropbox/Cerebro_development/pbmc_Seurat.crb'</span>,
  <span class="kw">experiment_name</span> <span class="kw">=</span> <span class="st">'pbmc_Seurat'</span>,
  <span class="kw">organism</span> <span class="kw">=</span> <span class="st">'hg'</span>,
  <span class="kw">groups</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'sample'</span>,<span class="st">'seurat_clusters'</span>,<span class="st">'cell_type_singler_blueprintencode_main'</span>),
  <span class="kw">cell_cycle</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'cell_cycle_seurat'</span>),
  <span class="kw">nUMI</span> <span class="kw">=</span> <span class="st">'nCount_RNA'</span>,
  <span class="kw">nGene</span> <span class="kw">=</span> <span class="st">'nFeature_RNA'</span>,
  <span class="kw">add_all_meta_data</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)</pre></body></html></div>
<pre><code>[10:03:13] Start collecting data...
[10:03:14] Overview of Cerebro object:

class: Cerebro_v1.3
cerebroApp version: 1.3.0
experiment name: pbmc_Seurat
organism: hg
date of analysis: 2020-02-19
date of export: 2020-09-25
number of cells: 5,697
number of genes: 15,907
grouping variables (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
cell cycle variables (1): cell_cycle_seurat
projections (2): UMAP, UMAP_3D
trees (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
most expressed genes: sample, seurat_clusters, cell_type_singler_blueprintencode_main
marker genes:
  - cerebro_seurat (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
enriched pathways:
  - cerebro_seurat_enrichr (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main, 
  - cerebro_GSVA (3): sample, seurat_clusters, cell_type_singler_blueprintencode_main
trajectories:
  - monocle2 (1): highly_variable_genes
extra material:
  - plots (1): dot_plot_marker_genes_clusters

[10:03:14] Saving Cerebro object to: ~/Dropbox/Cerebro_development/pbmc_Seurat.crb
[10:03:18] Done!</code></pre>
<p>Shown in the first screenshot is the interactive version of the plot:</p>
<p><a target="_blank" href="export_and_visualize_custom_tables_and_plots_files/plot_extra_material_2_interactive.png"><img src="export_and_visualize_custom_tables_and_plots_files/plot_extra_material_2_interactive.png" alt="Custom plot as extra material in Cerebro" width="100%" height="auto"></a></p>
<p>We notice that the legend for the dot size is missing. In cases where the plot isn’t shown correctly (conversion with plotly works well but not always perfectly), you can switch of interactivity and see the plain, unmodified plot:</p>
<p><a target="_blank" href="export_and_visualize_custom_tables_and_plots_files/plot_extra_material_2_plain.png"><img src="export_and_visualize_custom_tables_and_plots_files/plot_extra_material_2_plain.png" alt="Custom plot as extra material in Cerebro" width="100%" height="auto"></a></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Roman Hillje.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
